---
title: The effect of insulin on excitatory synaptic transmission in the rat dorsomedial
  hypothalamus
bibliography:
- "../Templates/thesis-citations.bib"
- ../Templates/packages.bib
csl: ../Templates/apa.csl
output:
  pdf_document:
    toc: false
    latex_engine: xelatex
    fig_caption: true
    includes:
      in_header: "../Templates/MtA-Thesis-Preamble.tex"
fontsize: 12pt
geometry:
- margin = 1in
- bindingoffset = 0mm # Set to 8 mm for printing
linestretch: 1.5
classoption: oneside
indent: true
editor_options:
  chunk_output_type: inline
always_allow_html: true
nocite: |
  @R-car, @R-broom, @R-dplyr, @R-extrafont, @R-ggforce, @R-ggfortify, @R-ggh4x, @R-ggpubr, @R-ggsignif, @R-ggtext, @R-glue, @R-heplots, @R-here, @R-kableExtra, @R-knitr, @R-MASS, @R-nlme, @R-patchwork, @R-rcompanion, @R-rstatix, @R-stringr, @R-tidyr, @R-viridis, @R-viridisLite
---

```{r set-global-options, echo=FALSE}
# This code achieves five things:

  # 1. Create plots of evoked excitatory currents versus time (P1 vs. Time) for individual cells and whole treatment groups
  # 2. Create a plot of the locations of all cells that you recorded from
  # 3. Run statistical tests
  # 4. Generate a thesis with auto-updating plots and p-values

# If this file is called Thesis.Rmd, you can knit the whole document into a PDF formatted according to MtA's Thesis guidelines.

# If this file is called Scripts-from-Thesis-Rmd.R, it contains only the code chunks within the thesis.
# Don't make any changes to Scripts-from-Thesis-Rmd.R because it will be overwritten when running purl() in Thesis.Rmd.


# ------------ Set chunk options --------------------------------------------------------------
# cairo_pdf encodes high-quality plots with searchable text
# Hide chunk warnings to reduce distractions
# comment = NA removes the double hash tags next to chunk output

knitr::opts_chunk$set(
  dev = c("cairo_pdf"),
  dpi = 600,
  fig.align = "center",
  comment = NA,
  echo = FALSE,
  message = FALSE,
  warning = FALSE
)
```

```{r load-libraries, include = FALSE}
# ------------- Load packages ----------------------------------------------------------------------------
# In a .Rmd file, the default working directory is the folder that contains the .Rmd file
# However, in an .R file, the default working directory is the parent directory i.e. the project folder R-work/
# Use here() to define relative paths and ensure that everything is relative to the top-level directory
# Otherwise, R will not be able to find files when you are working with both Thesis.Rmd and .R script files
# Use here() anytime you are reading in an external file

# here() also makes it easier to transfer between different computers
# because each person's computer has a different file structure
# For example, instead of C:/Users/cslaureijs/Documents/R-work/Thesis/Thesis.Rmd, which only works on my computer
# You can just type here("Thesis/Thesis.Rmd")

library(here)
library(abftools) # Required to read in .ABF files

# Data Management

library(tidyr) # Reformat data frames from wide to long and vice versa 
library(dplyr) # Data manipulation
library(stringr) # Manipulating text objects
library(glue) # Required for easy mixing of variables and text when naming objects

# Making plots

library(ggplot2) # Make plots
library(viridis) # Accessible and beautiful colour palettes
library(extrafont) # Required for custom fonts in plots
library(grid) # Required for rasterGrob function to have a background image on a plot
library(gridExtra) # Required for more rasterGrob functions on cell coordinates plot
library(ggtext) # Required for formatting plot text (e.g. coloured text in title)
library(ggforce) # Required for sina plots
library(patchwork) # Required for multi-plot layout
library(ggsignif) # Required for significance stars and brackets on plots
library(ggh4x) # Required to create nested x-axis labels
library(ggpubr) # Required for significance brackets

# Statistical packages

library(ggfortify) # Autoplot multiple diagnostic plots at once
library(nlme) # Required for linear mixed models and other statistical tests
library(MASS) # Get studentized residuals for linear regression
library(rcompanion) # Plot residuals against a normalized histogram
library(car) # Provides ncvTest and other functions
library(heplots) # Required for Box's M test for MANOVA
library(RVAideMemoire) # Required for multivariate normality tests
library(broom) # Required to format statistical test output into tidy dataframes
library(rstatix) # Required to use pipes with statistical tests
library(knitr) # Required to access better table formatting through kable
library(kableExtra) # Required for full width tables
library(lazyWeave) # Provides the pvalString() function to format publication-ready p-values in a table 


# Redefine the function select() to prioritize dplyr select()
# This is because the MASS select() function masks dplyr's select()

select <- dplyr::select


# Generate a bibliography of all packages used in this file
# I will use this later to cite all packages that I used in my bibliography

knitr::write_bib(file = here("Templates","packages.bib"))
```

```{r load-functions, include = FALSE}
# ------------- Load scripts ---------------------------------------------------------------------------
# This script file contains my functions for pruning the data and creating plots of P1 vs. Time
# I sourced these to an external file to reduce the length of my code

# Set local = knitr::knit_global() to ensure that the script uses THIS document's directory,
# not the Scripts/ folder

source(here("Scripts/Functions.R"), local = knitr::knit_global())
```

```{r set-plot-options}
# ---------- Set plot theme, colours & sizes --------------------------------------------------------------------------------
# Set save_choice to "yes" if you want to save plots (! this will increase the run time; generally only run this after adding new data)
# Saved plots will automatically go to Figures/Output-... folders depending on what type they are
  # E.g. Raw plots of the first evoked current over time (P1 vs. Time) will go into Figures/Output-individual-plots

save_choice <- "no"


treatment_shapes <- c(16, 17, 15, 18)
list_of_treatments <- c("Control", "Fasting", "HNMPA", "PPP")
list_of_significance_stars <- c(
  "***" = 0.001,
  "**" = 0.01,
  "*" = 0.05,
  "ns" = 2
)

# Set colours here only for consistency

line_col <- "#333333" # Sets colour of x and y-axes
my_colours <- c("#6600cc", "#0093fb", "#55b323","#ffe70f","#e86c00","#333333", "#411900","#e11584")
my_colours_pale <- c("#b080e0","#5cb9fa", "#92d46e","#ebdf05","#f58c31","#b1b1b1","#a16b4a","#f57dbe")

# Rectangle highlights intervals from 5-10 min and 15-20 min
rectangle_shading_colour <- "#f6f6f6"

# Required for better contrast between the sexes in summary plots
# Do NOT use these for any of the raw P1 vs. Time plots because they are too pale and/or too dark
my_colours_very_dark <- c("#4d0299","#026bb5","#398511","#a69502","#994700","#000000","#785138","#910150")
my_colours_very_pale <- c("#d6b8f5", "#8fd0ff", "#aee691","#ebdf05","#ffc38f","#dcdcdc","#bf9b84","#eba2c9")

# Custom fonts may cause issues depending on what fonts you have in your system
# Troubleshooting steps: 
  # Try changing the font to one that you have on your computer
  # If it does not work, you could always delete 'family = plot_font_family' in the ggplot theme set below
plot_font_family <- "Segoe UI"
plot_light_font_family <- "Segoe UI Light"
significance_stars_font <- plot_font_family


# Set default colours and point sizes for action potential summary plots
mean_point_colour <- "#000000"
baseline_group_colour <- "#727a85"
insulin_group_colour <- "#6600cc"
connecting_line_width <- 0.1
connecting_line_colour_aps <- "#bfbfbf"
connecting_line_width_PPR <- 0.2

mean_point_size <- 0.5
geom_sina_size <- 3
geom_signif_text_size <- 8

AP_trace_size <- 0.7
scale_bar_shift_y <- 5
scale_bar_shift_x <- 30

# These sizes work better for the multi-plot figure in the PDF output
if (knitr::is_latex_output()) {
  geom_sina_size <- 2
  geom_signif_text_size <- 4
  AP_trace_size <- 0.6
  scale_bar_shift_y <- 13
  scale_bar_shift_x <- 200
  connecting_line_width <- 0.04
  connecting_line_width_PPR <- 0.04
}

# A consistent y-axis enables comparison across multiple experiments, treatments, etc.
# This is not applied to the raw plots of eEPSCs vs. time for individual cells
# If you change this from 175, make sure to regenerate all summary plots so that you can compare across the same y-axis.

y_axis_limit <- 175
PPR_y_min <- 0
PPR_y_max <- 5
```

```{r set-ggplot2-theme}
# ----------- Set ggplot theme ----------------------------------------------------------------------------------
# Formatting changes like increasing font size & removing gray background
# Requires the extrafont() package (loaded in the load-libraries chunk) for custom font

theme_set(
  theme_classic() %+replace%
    theme(
      text = element_text(
        family = plot_font_family
      ),
      plot.title = element_text(
        color = "black",
        size = 20,
        family = plot_font_family,
        #face = "plain",
        margin = margin(b = 25),
        hjust = 0.5
      ),
      plot.margin = margin(25, 25, 25, 25),
      plot.caption = element_text(
        hjust = 0,
        family = plot_font_family,
        size = 12
      ),
      plot.caption.position = "plot",
      axis.text = element_text(
        size = 12,
        color = "black"
      ),
      axis.title = element_text(
        size = 16,
        face = "bold"
      ),
      axis.title.y = element_text(
        margin = margin(r = 25),
        angle = 90,
        vjust = 0.5
      ),
      axis.title.x = element_text(margin = margin(b = 25, t = 20)),
      axis.ticks = element_blank(),
      strip.background = element_rect(color = NA, fill = NA),
      strip.text = element_text(size = 20)
    )
)
```

```{=tex}
\begin{centering}
\begin{spacing}{1}
\Large

by

\LARGE

Christelinda Laureijs

\end{spacing}
\end{centering}
```
```{=tex}
\large
\vspace*{\fill}
```
<!-- Although it is not ideal, I am manually setting the linebreaks here. This is to conform to MtA's title page standards -->

```{=tex}
\begin{centering}
\begin{spacing}{1}
A thesis submitted to the\\
Department of Biology\\
Mount Allison University\\
in partial fulfillment of the requirements for the\\
Bachelor of Science degree with Honours\\
in Biology

April 10, 2024

\end{spacing}
\end{centering}
```
```{=tex}
\newpage
\vspace*{\fill}
\large
```
\adforn{20} *COLOPHON* \adforn{48}

\normalsize

This thesis was typeset in R Markdown and the text was set in EB Garamond. The data and code used to produce this document are available at [*https://github.com/christelinda-laureijs/honours-thesis*](https://github.com/christelinda-laureijs/honours-thesis).

```{=tex}
\newpage
\pagenumbering{roman}
```
# Abstract

Insulin plays a key role in regulating blood glucose levels and it also acts in brain regions involved with appetite regulation. The dorsomedial hypothalamus (DMH) is one such brain region, and although it is a major site of insulin receptors, virtually nothing is known about the effect of insulin on neuronal activity and communication between DMH neurons. Since DMH neurons stimulate appetite and insulin is a satiety hormone, I hypothesized that insulin inhibits both neuronal activity and neuronal communication in the DMH. I used whole-cell patch clamp electrophysiology to record from living DMH neurons taken from young male and female Sprague-Dawley rats. I compared recordings before and after applying 500 nM of insulin [@labouebeetal2013] to determine the effect of insulin on action potentials (a measure of neuronal excitability) and excitatory synaptic transmission (a measure of neuronal communication). Insulin significantly decreased action potential frequency and excitatory current amplitudes in both sexes, indicating that insulin decreases neuronal excitability and excitatory synaptic transmission in the DMH. Recordings taken with various insulin receptor blockers present suggested that insulin binds to insulin-like growth factor 1 receptors. Insulin signalling in the DMH is also independent of feeding state, suggesting that it is involved with regulating physiological processes beyond appetite regulation. This research contributes to our understanding of how insulin acts in the brain to influence energy metabolism, with direct applications to research on the pathophysiological effects of insulin resistance in obesity and diabetes.

```{=tex}
\setcounter{tocdepth}{2}
\tableofcontents
```
\listoffigures

# List of Abbreviations

DMH \tab{Dorsomedial hypothalamus}\
GABA \tab{Gamma-aminobutyric acid}\
eEPSC \tab{Evoked excitatory post-synaptic current}\
PPR \tab{Paired pulse ratio}\
HNMPA \tab{Hydroxy-2-naphthalenylmethylphosphonic acid}\
PPP \tab{Picropodophyllotoxin}

```{=tex}
\cleardoublepage
\normalsize
\pagenumbering{arabic}
```
# Introduction

Obesity is a serious health concern in New Brunswick, where the number of adults with obesity is 11% higher than the national average of 26.8% [@statisticscanada2019]. Obesity is an imbalance in energy intake and energy expenditure that results in excess energy levels in the body [@berthoud&morrison2008]. Although most excess energy is stored as fat, long-term periods of excess energy also negatively impact metabolic processes and increase the risk of developing other health conditions [@berthoud&morrison2008], including asthma, type II diabetes, cardiovascular disease, and various cancers [@sarmaetal2021]. Understanding the role of hormones in regulating appetite could lead to the development of new obesity therapies.

## Hormones

Hormones are critical for regulating a diverse range of physiological processes such as growth, reproduction, and appetite. When animals are hungry, the stomach releases hormones like ghrelin, which increases the activity of appetite-stimulating neurons to create a desire to find and consume food [@wrenetal2000]. Once an animal has eaten, a new series of hormones enter the brain where they inhibit appetite-stimulating neurons and activate neuronal pathways that induce satiety [@wilding2002; @woodsetal2006]. Through their close interactions with digestive system organs and the brain, hormones can respond quickly to short-term energy needs.

Hormones also influence appetite over the long term in response to an animal’s energy state. Leptin is one example of a hormone that acts as an indicator of stored energy in the body [@considineetal1996]. Since adipocytes (fat-storing cells) produce and release leptin, higher bloodstream concentrations of leptin correlate to more body fat, and in turn, more stored energy [@considineetal1996]. Leptin stimulates neurons to produce neurotransmitters that eventually decrease appetite [@eliasetal1999]. A second long-term energy indicator is insulin [@wilding2002], which is also a key energy regulator.

## Insulin

Insulin is a hormone that regulates blood glucose levels, digestive processes, and body weight [@demeyts2000]. After a meal, the pancreas releases insulin into the bloodstream, where it activates molecular pathways that lower blood glucose levels [@demeyts2000]. Insulin also interacts with digestive organs to influence metabolic pathways that maximize an animal’s available energy resources. Insulin carries out these functions by binding to insulin receptors or insulin-like growth factor-1 receptors and activating downstream molecular pathways (Figure \ref{insulin-pathway}). In addition to acting on body cells, insulin also acts in the brain.

```{r insulin-schematic}
#| fig.cap = "When insulin binds to a cell, it activates a series of molecular pathways involved in energy metabolism and gene expression. Insulin can bind to either an insulin receptor (IR) or an insulin-like growth factor 1 receptor (IGF-1R). The phosphatidylinositol 3-kinase (P13K) pathway is primarily involved in regulating metabolic processes such as glucose metabolism, lipid metabolism, and gluconeogenesis, while the mitogen activated protein kinase (MAPK) pathway is primarily involved with regulating gene expression and cell division (De Meyts, 2000). Parts of the P13K pathway can affect synaptic transmission and neuronal activity, with mechanistic target of rapamycin (mTOR) playing a key role in synaptic transmission (McCabe et al., 2020). Although it looks like there are two distinct pathways, the downstream components of the P13K and MAPK pathways frequently interact with one another (De Meyts, 2000). \\label{insulin-pathway}",
#| fig.scap = "Insulin binding activates a series of molecular pathways.",
#| out.width = "100%",
#| purl = FALSE


# Note chunk options in Thesis.Rmd
# fig.cap = long caption that will print below the figure
# Caption includes \\label{label}, which enables in-text referencing
  # As you can see in Figure \ref{label}...

# fig.scap = short caption that LaTeX will use in the List of Figures
# out.width = 100% of line width
# Excluded from Scripts-from-Thesis-Rmd.R (purl = FALSE) because this is for the thesis only

knitr::include_graphics(here("Figures/Insulin-schematic.png"))
```

Within the brain, insulin binds to insulin receptors on neurons and alters their activity [@havrankovaetal1981]. Insulin can activate neurons or inhibit them, depending on the signalling pathways that are activated, the type of neuron involved, and/or the location of the neuron. In some neurons, insulin binding activates changes within neurons that make them more easily activated, such as increasing receptors for excitatory neurotransmitters [@liuetal1995] or increasing intracellular Ca^2+^ concentrations [@jonasetal1997]. In other cases, insulin stimulates neurons to express receptors for neurotransmitters like gamma-aminobutyric acid (GABA; @wanetal1997), which makes them more easily inhibited. Through its interactions with insulin receptors, insulin affects individual neurons, which then impacts entire neuronal pathways.

Most brain regions that respond to insulin help animals to find, consume, and enjoy food. The olfactory bulb and the hypothalamus contain the highest numbers of insulin receptors [@corpetal1986; @havrankovaetal1981], making them more sensitive to insulin than other regions. Not only does the olfactory bulb assist with locating and identifying food, but it also influences meal choices, salivation, and appetite [@gaillet-torrentetal2014; @proserpioetal2017]. Within the hypothalamus, subdivisions called nuclei regulate a diverse range of physiological processes relating to appetite and energy metabolism. Insulin can also amplify leptin signalling in the hypothalamus [@carvalheiraetal2001], suggesting that insulin works with other hormones to inhibit appetite-stimulating hypothalamic pathways. These findings demonstrate that insulin is a satiety hormone that acts on neurons in the hypothalamus to influence a wide variety of processes relating to food intake. Insulin also acts on neurons in regions outside of the hypothalamus. One example is the ventral tegmental area, a brain region involved in reward and pleasure [@labouebeetal2013]. All these findings demonstrate that insulin plays an important role in regulating appetite, energy metabolism, and the rewarding feelings associated with eating.

If insulin is so closely linked to food intake, how does it respond to an animal’s energy needs? Insulin could act as a marker of feeding state. Since insulin levels in the bloodstream fluctuate with meals and digestion, this could allow for precise changes in neuronal activity in response to an animal’s current energy state. Surprisingly, insulin concentrations in the brain are relatively stable compared to the bloodstream [@havrankovaetal1981]. This implies that the brain maintains an independent insulin supply [@gerozissisetal1997], either by synthesizing insulin locally or by storing insulin as it enters the brain [@havrankovaetal1981]. The independence of brain insulin levels from the rest of the body suggests that consistent insulin levels are required for neuronal function, not just appetite regulation. Some insulin from the bloodstream does enter the brain, however, and injecting insulin into the bloodstream decreases appetite [@vanderweele1994]. Insulin likely enters the brain through a structure called the median eminence, where it then has immediate access to the hypothalamus and nearby brain regions [@romanoetal2023].

## The Dorsomedial Hypothalamus

The dorsomedial hypothalamus (DMH) is a hypothalamic nucleus that regulates appetite [@daltonetal1981] and body weight [@bellingeretal1979]. Neurons in the DMH express receptors for appetite-regulatory hormones including leptin [@zhangetal2011] and insulin [@corpetal1986]. DMH neurons also connect to many brain regions involved with appetite regulation and nutrient sensing [@horst&luiten1986; @thompson&swanson1998]. Figure \ref{DMHfig} summarizes the major projections to and from the DMH.

Animals require an intact DMH for appetite regulation. When researchers destroyed neurons in the DMH with an electric current, animals experienced decreased food intake and body mass [@bellingeretal1979; @daltonetal1981]. By sequentially cutting neuronal pathways to and from the DMH, researchers found that DMH neurons require posterior and ventral projections to interact with appetite-regulatory neuronal pathways [@bellinger&bernardis1999knife]. Researchers also learned that DMH neurons are more active after fasting [@groessletal2013], and before and after meals [@angeles-castellanosetal2004]. By selectively "turning on" DMH neurons through optogenetic activation, researchers can increase food intake in mice [@otgon-uuletal2016; @jeongetal2017]. These studies all show that the DMH is vital for appetite regulation.

```{r Include-figure-DMH}
#| fig.cap = "The DMH interacts with many brain regions. Two particularly important regions include the nucleus of the solitary tract (NST) and parabrachial nucleus (PB), which help with nutrient sensing. The DMH has projections leading to and from other brain regions that control appetite, such as the arcuate nucleus (ARC), ventromedial hypothalamus (VMH) and lateral hypothalamus (LHA). Dashed lines indicate locations where neuronal projections cross to the other half of the brain. This figure is based on the neuronal tracing studies of Horst \\& Luiten (1986) and Thompson \\& Swanson (1998). \\label{DMHfig}",
#| fig.scap = "The DMH interacts with many brain regions",
#| out.width = "100%",
#| purl = FALSE

knitr::include_graphics(here("Figures/DMH-Projections.png"))
```

Although insulin interacts with neurons in the hypothalamus [@corpetal1986] and DMH neurons express insulin receptors [@corpetal1986], nothing is known about the effect of insulin on DMH neurons. It is also unclear how insulin binds to DMH neurons, and how it could affect neuronal activity.

## The Current Study

The main question of this study is: Does insulin bind to DMH neurons, and if so, how does it affect their activity? Since DMH neurons stimulate appetite, and insulin decreases appetite, I hypothesize that insulin decreases neuronal activity and communication within the DMH. I predict that insulin decreases action potential frequency and current amplitudes associated with excitatory neurotransmitters (a measure of neuronal communication) in DMH neurons relative to control conditions. To see if insulin interacts with DMH neurons, I recorded from living DMH neurons and compared action potential parameters and excitatory current amplitudes before and after adding insulin. I also compared neuronal activity in the DMH neurons of rats who underwent a 24-hour fasting protocol to see if insulin signalling in the DMH varies with feeding state. This research increases our understanding of appetite regulation in the brain, with direct applications to research on the pathophysiological effects of conditions like obesity and diabetes.

\cleardoublepage

# Methods

## Animals

I used young (28-44 days old) male and female Sprague-Dawley rats from Charles River Laboratories (Québec, Canada) for all treatment groups. The rats arrived when they were 22 days old, and they had a one-week acclimation period. The rats lived in groups of 4 with free access to standard rat chow and water. All cages had wood shavings for bedding, toys (wooden blocks and a plastic chew toy) for enrichment and Crawl Balls™ (Bio-Serv) for shelter. The animal rooms were maintained at 21$^{\circ}$C and 50% $\pm$ 10% humidity, with a 12-hour light/dark cycle. The Animal Care Committee at Mount Allison University approved the project (Protocol #103088).

## Brain Removal & Slice Preparation

I placed a rat into an anesthetization chamber filled with 5% isoflurane gas in oxygen. After confirming that the rat was unconscious through a toe pinch test, I decapitated it with a guillotine (World Precision Instruments). I transferred the brain to an ice-cold, freshly oxygenated slicing solution containing (in mM) 87 NaCl, 2.5 KCl, 25 NaHCO~3~, 0.5 CaCl~2~ $\cdot$ 2H~2~O, 7 MgCl~2~ $\cdot$ 6 H~2~O, 1.25 NaH~2~PO~4~, 25 glucose and 75 sucrose. I had previously bubbled the solution with carbogen (5% carbon dioxide in oxygen) for 10-15 minutes to ensure full oxygen saturation.

I removed the brain from the solution and cut off the anterior, posterior, and dorsal portions. The brain was glued posterior side down (Krazy Glue) into a vibrating microtome (Leica Biosystems) next to a piece of agar for support. It was then re-immersed in slicing solution with carbogen bubbling. The microtome created thin slices (250 $\mu$m) along the coronal plane, while I identified brain regions with an atlas [@paxinos&watson2009].

On a DMH-containing brain slice, I used a scalpel to remove the lateral edges and then cut the slice along the midline. I transferred the resulting hemisections to a beaker containing artificial cerebrospinal fluid. This fluid contained (mM) 126 NaCl, 2.5 KCl, 26 NaHCO~3~, 2.5 CaCl~2~ $\cdot$ 2H~2~O, 15 MgCl~2~ $\cdot$ 6 H~2~O, 1.25 NaH~2~PO~4~, and 10 glucose. A gas line kept the artificial cerebrospinal fluid bubbled with carbogen, and a water bath maintained the solution at 32$^{\circ}$C. I let the slices recover for one hour before recording to ensure maximum cell health.

## Recording

I placed a hemisection in a perfusion chamber filled with continuously oxygenated artificial cerebrospinal fluid (32$^{\circ}$C; TC-344C Temperature Controller, Warner Instruments) flowing through at 1 mL/minute. The chamber was mounted under an upright microscope (Olympus BX51WI) connected to an Infinity 2 camera and software (Version 6.5.2, Lumenera Corporation), which projected to a computer monitor. I navigated to the DMH, which was located along the border of the third ventricle [@paxinos&watson2009].

I used whole-cell patch clamp electrophysiology to stimulate DMH neurons and record their responses in real-time. The neurons were still alive, so this technique enabled me to observe changes in activity after experimental manipulations. During patch clamping, two pipettes were positioned in the brain slice: a stimulating pipette and a recording pipette (Figure \ref{pipette-figure}). The stimulating pipette used electrical stimuli to stimulate nearby neurons to release neurotransmitters onto the target neuron. The recording pipette recorded the resulting currents from the target neuron.

```{r Include-figure-pipettes}
#| fig.cap = "\\emph{Left}: In whole-cell patch clamp electrophysiology, the stimulating electrode (dashed blue) stimulates nearby neurons (pale yellow) to release neurotransmitters while the recording electrode (green) records from the target neuron (yellow). \\emph{Right}: The recording electrode is inserted directly into a living DMH neuron (40X magnification). \\label{pipette-figure}",
#| fig.scap = "Whole-cell patch clamp electrophysiology uses two electrodes to manipulate and record from living neurons",
#| out.width = "100%",
#| purl = FALSE

knitr::include_graphics(here("Figures/Methods-Patch-Clamp-Technique.png"))
```

I used pipettes with resistances ranging from 4.0 to 6.0 M$\Omega$s (Model P-1000 pipette puller, Sutter Instruments). The stimulating pipette contained artificial cerebrospinal fluid, while the recording pipette had internal solution, consisting of (mM) 108 KGluconate, 8 KCl, 8 NaGluconate, 10 HEPES, 2 MgCl~2~, 1 K~2~ETGA, 4 K~2~ATP, 0.3 NA~3~GTP, which simulated the intracellular fluid within neurons. The internal solution was adjusted to pH 7.2 with 1 M KOH and 285-300 mOsm with sterilized water.

This study focused on currents associated with the excitatory neurotransmitter glutamate because DMH neurons receive a large number of excitatory inputs and glutamate-mediated transmission plays a large role in regulating DMH activity [@baileyetal2003]. Since DMH neurons also receive inhibitory inputs through the inhibitory neurotransmitter GABA [@baileyetal2003], I added 50 $\mu$M picrotoxin (Tocris Bioscience) to the perfusion solution. Picrotoxin blocked GABA~A~ receptors [@newland&cull-candy1992] and allowed me to isolate and record glutamate currents only.

I lowered both pipettes into the brain slice using micromanipulators (MPC-365 System, Sutter Instruments). I inserted positive pressure into the recording pipette to blow away debris and to keep the tip clean. Once I identified a neuron, I gently pushed the recording pipette against the cell membrane until a dimple formed. Releasing the positive pressure caused the cell to form a seal against the pipette tip. I applied a small pulse of mouth suction to break through the membrane and gain access to the cell.

I used pCLAMP software (Version 10.6.0.13, Molecular Devices) to perform protocols and record the resulting electrophysiological parameters. A MultiClamp700B (Molecular Devices) amplified the neuron's electrical signals and then a Digidata 1550B (Molecular Devices) low-pass filtered and digitized them at 1 kHz and 10 kHz, respectively. I monitored the cell's stability every 5 minutes using access resistance and discarded any recordings with unhealthy or lost cells.

## Action Potentials

I used a current clamp steps protocol to observe action potentials and compare their properties. I performed this protocol immediately with each new cell as a baseline, then recorded evoked excitatory currents for 25 minutes after adding insulin (see the *evoked currents* section for a protocol summary). After I finished the evoked currents protocol, I recorded the current clamp steps protocol once again. During the current clamp steps protocol, the software stimulated the cell for 500 ms with electrical current injections of increasing magnitude. The current injections began at -50 pA, and increased by 10 pA with each step until a maximum of 40 pA. The current injections depolarized the neuron to a point where it fired action potentials.

I identified the first current injection that resulted in action potentials and then selected the first action potential for my analysis (Figure \ref{ap-schematic}). I identified the action potential threshold (the depolarization required to induce an action potential; in mV) through the first derivative method [@farriesetal2010]. This method sets the threshold as the membrane potential where the action potential velocity is 10 mV/ms or greater [@farriesetal2010]. The latency to fire was the time (in ms) between the start of the stimulus and the action potential's peak amplitude.

All other action potential properties were relative to the threshold. I measured the action potential's amplitude (mV) and half-width (ms; Figure \ref{ap-schematic}). I also measured the after-hyperpolarization amplitude (mV) and time of the after-hyperpolarization (ms). Lastly, I counted the number of action potentials at each current injection and used these values to obtain the action potential frequency per current injection (Hz). If the recording contained no action potentials, I set the action potential frequency and amplitude to 0, and set all other action potentials to *NA* values. The threshold, amplitude, frequency, and all other action potential properties conveyed information about the intrinsic excitability of a neuron. For example, the half-width (a measure of the width of the action potential measured at half the amplitude) is an indicator of the repolarization rate [@bean2007].

By comparing action potential properties before and after insulin exposure, I was able to explore the effect of insulin on intrinsic neuronal excitability. Since the action potential properties data did not pass the assumptions of normality, even with a log-transformation, I used a paired Wilcoxon signed-rank test to assess differences in action potential parameters across the two states.

```{r Include-figure-ap-schematic}
#| fig.cap = "A schematic of the current clamp steps protocol. \\emph{Top:} This is a representative recording from a single cell featuring all 10 current clamp steps stacked on top of one other. The current injections ranged from -50 pA (bottom; orange) to 40 pA (top; purple), with a 10 pA increase per step. I used the first action potential at the lowest current injection (boxed outline at a current injection of 20 pA) to analyze action potential properties. All traces have been smoothed through node reduction using the \\emph{simplify path} function in Inkscape. \\emph{Bottom:} I measured the action potential threshold, amplitude, half-width, and latency to fire. I also measured the after-hyperpolarization amplitude and time of the after-hyperpolarization. \\label{ap-schematic}",
#| fig.scap = "The current clamp steps protocol with labelled action potential properties",
#| out.width = "100%",
#| purl = FALSE

knitr::include_graphics(here("Figures/AP-Schematic.png"))
```

## Evoked Currents

During the evoked currents protocol, the software maintained the cell at a membrane potential of -70 mV, which is close to a DMH neuron's natural resting membrane potential of -60 mV [@baileyetal2003]. Through the stimulating pipette, the software delivered two electrical stimuli 50 ms apart, repeated every 5 seconds (Figure \ref{evoked-currents-schematic}). The stimuli induced nearby neurons to release glutamate onto the target cell, which then experienced a temporary change in membrane potential. These currents are called evoked excitatory post-synaptic currents (eEPSCs, or "evoked currents") because they are a) artificially induced, b) caused by an excitatory neurotransmitter, and c) present in the cell that receives the neurotransmitters, not the presynaptic cell.

I recorded evoked excitatory currents for 5 minutes under control conditions as a baseline. I then added 500 nM insulin to the perfusion solution and continued recording for 25 minutes. The concentration of insulin applied during the recordings is much higher than the 0.16 nM hypothalamic insulin concentration reported by Baskin et al. (1983). However, this concentration is comparable to past work examining excitatory synaptic transmission in other brain areas [@labouebeetal2013; @wang&linden2000]. I also applied insulin in the presence of an insulin receptor blocker (Hydroxy-2-naphthalenylmethylphosphonic acid; HNMPA; 300 $\mu$M; Abcam Incorporated) and an insulin-like growth factor 1 receptor blocker (picropodophyllotoxin; PPP; 0.5 $\mu$M; Selleck Chemicals) to explore insulin's binding mechanisms.

To determine if insulin signalling in the DMH varies with feeding state, I placed a rat alone into a cage with no food for 24 hours before brain removal. During this period, the rat had free access to water, toys for environmental enrichment, bedding, and shelter. Removing food creates a hunger state, which would potentially alter insulin signalling at the DMH.

I analyzed the evoked current data for all treatments. I started by normalizing the first evoked current amplitude as a percentage of the cell's mean baseline evoked current amplitudes (100%) for easy comparison across cells with different starting amplitudes. Then, I used a multivariate repeated measures ANOVA to assess changes in current amplitudes over time. The model used a within-subjects factor of time (six 5-minute intervals) and treatment and sex as between-subjects factors. Later, I split the data by treatment and used paired t-tests with Holm's corrections [@holm1979] to identify the time when the currents became significantly different than the baseline. I assessed normality using QQ plots and the Shapiro-Wilk normality test or its multivariate equivalent. I used Levene's test or Box's M test (multivariate equivalent) to assess the homogeneity of variances.

```{r include-figure-evoked-currents}
#| fig.cap = "For the evoked currents protocol, the cell experienced two electrical stimuli 50 ms apart, repeated every 5 seconds. This created evoked currents, which I recorded and compared. I measured changes in the amplitude of the first evoked current (blue) over time. I also analyzed changes in the paired pulse ratio (PPR), which is the ratio of the second evoked current to the first evoked current. If the PPR changed throughout the recording, it indicated that the change in evoked current amplitude occurred through a presynaptic mechanism (Manita et al., 2007). \\label{evoked-currents-schematic}",
#| fig.scap = "A schematic of the evoked currents protocol.",
#| out.width = "100%",
#| purl = FALSE
knitr::include_graphics(here("Figures/Evoked-Currents-Schematic.png"))
```

## Paired Pulse Ratio

The paired pulse ratio (PPR) is the ratio between the second and first evoked currents (Figure \ref{evoked-currents-schematic}). Since the PPR is a measure of the probability of neurotransmitter release [@manitaetal2007], changes in the PPR over time can provide information about the mechanism of synaptic transmission. If the PPR changes throughout the recording, this indicates that a presynaptic mechanism had changed. If the PPR does not change, this indicates that a change had occurred post-synaptically.

Using the raw data (i.e. the non-normalized data), I divided the amplitude of the second evoked current by the first evoked current to obtain the PPR. I then filtered the PPRs to values within a range of 0 to 5 and averaged the PPR across the baseline period (0 to 5 minutes) and final interval (20 to 25 minutes) within each cell and used a paired T-test to compare the PPR across these two periods.

## Software

I measured electrophysiological parameters like evoked current amplitude and action potential properties using *Clampfit* (Version 10.6.0.13, Molecular Devices). I performed all statistical analyses with *R Statistical Software* (`r paste0("Version ",substr(version$version.string,11,15))`; R Core Team 2021) in *RStudio* (Build 369; @rstudioteam2020). I used $\alpha$ = 0.05 for all tests.

I used the *abfTools* package (Version `r packageVersion("abftools")`; @wu2019) to obtain representative traces from raw recording files, and I made all plots with *ggplot2* (Version `r packageVersion("ggplot2")`; @wickham2016a). I used *Inkscape* (Version 1.3; @inkscapedevelopers2023), an open-source vector editor, to create the explanatory schematics. I also used Inkscape to remove stimulus artifacts on the representative traces on Figures \ref{summary-plot-control}, \ref{summary-plot-HNMPA}, \ref{summary-plot-PPP}, and \ref{summary-plot-fasting}.

```{r Include-figure-methods}
#| fig.cap = "A visual summary of the experiment. I removed rat brains, stimulated DMH neurons, and recorded the resulting currents. This allowed me to compare currents before and after exposing the neurons to insulin. I also compared action potentials before and after adding insulin. \\label{methods-figure}",
#| fig.scap = "I removed rat brains, recorded from DMH neurons, and compared recordings before and after insulin exposure",
#| out.width = "100%",
#| purl = FALSE

knitr::include_graphics(here("Figures/Methods-Schematic-with-APs.png"))
```

```{r AP-data-import-and-merge, include=F}
# --------------- AP Analysis --------------------------------------------------------------------------------------
ap_data <- read.csv(here("Data/Raw-CSVs/20240329-AP-Analysis.csv"), header = T)

# Remove recordings with known issues
# AH, AM (Current clamp steps 'After' recording was after Insulin and high-frequency stimulation)
# FO, FP (Lost the cell before I could record 'After') 

ap_data <- ap_data %>%
  filter(! Letter %in% c("AH", "AM", "FO", "FP"))

# Add new column with frequency of APs
# Each sweep lasts 0.5 seconds, so to get frequency in Hz, I divided the number of APs by 0.5

# Add a new column that states the current injection used at each step
# Rename column 't_x' to 'Threshold', which is more useful

ap_data <- ap_data %>%
  mutate(
    AP_frequency = No_of_APs / 0.5,
    Latency_to_fire = Time_to_Peak - 265.4,
    Antipeak_time_relative_to_threshold = Time_of_antipeak - Time_of_Threshold,
    Current_injection = case_when(
      Sweep == 1 ~ -50,
      Sweep == 2 ~ -40,
      Sweep == 3 ~ -30,
      Sweep == 4 ~ -20,
      Sweep == 5 ~ -10,
      Sweep == 6 ~ 0,
      Sweep == 7 ~ 10,
      Sweep == 8 ~ 20,
      Sweep == 9 ~ 30,
      Sweep == 10 ~ 40,
      T ~ 50
    )
  ) %>%
  rename(Threshold = t_x)

# Import CSV with info on sex, synapses, X, Y, treatment, animal, etc.
cell_characteristics <-
  read.csv(here("Data/Plaintext-Cell-Characteristics.csv"))

# Merge datasets to get complete AP dataset
full_ap_df <-
  merge(ap_data,
        cell_characteristics,
        by = "Letter")

# Create list of the order of AP parameters
# This ensures that the parameters are listed in a consistent order in the t-test and Shapiro-Wilk tables
list_of_AP_parameters <-
  c(
    "Peak_amplitude",
    "Threshold",
    "Half_width",
    "Latency_to_fire",
    "Antipeak_amplitude",
    "Antipeak_time_relative_to_threshold"
  )
```

```{r AP-data-get-representative-traces}
# ----------------- AP Analysis: Get representative current clamp steps traces --------------------

# Set range (e.g. height and width) of scale bar
scale_bar_x_in_recording_time_units <- 500
scale_bar_y_in_mV <- 20

# Manually re-position the text labels on the scale bar
scale_bar_shift_y <- 10
scale_bar_shift_x <- 40

# Set starting point of the scale bar
# This determines the point where the horizontal and vertical bars meet
scale_bar_start_x <- 8200
scale_bar_start_y <- 0

# Set the x- and y-axis limits
# The x-limit shouldn't have to change much since all protocols started and ended at the same time
ap_traces_y_min <- -120
ap_traces_y_max <- 20
ap_traces_x_min <- 2500 #2500
ap_traces_x_max <- 8500 #8000


ap_trace_baseline <-
  make_AP_trace_plot(file_ID = "23623001_rescaled_0_01",
                     sweep1 = "epi10",
                     sweep2 = "e",
                     trace_color = baseline_group_colour) +
  annotate(
    "segment",
    x = scale_bar_start_x,
    xend = scale_bar_start_x + scale_bar_x_in_recording_time_units,
    y = scale_bar_start_y,
    yend = scale_bar_start_y,
    lwd = 0.4
  ) +
  annotate(
    "segment",
    x = scale_bar_start_x,
    xend = scale_bar_start_x,
    y = scale_bar_start_y-0.49,
    yend = scale_bar_start_y + scale_bar_y_in_mV,
    lwd = 0.4
  ) +
  annotate(
    "text",
    x = scale_bar_start_x - scale_bar_shift_x,
    y = scale_bar_start_y + 1/2*(scale_bar_y_in_mV),
    label = paste0(scale_bar_y_in_mV, " mV"),
    hjust = 1,
    vjust = 0.5,
    family = plot_font_family
  ) +
  annotate(
    "text",
    x = scale_bar_start_x + 1/2*(scale_bar_x_in_recording_time_units),
    y = scale_bar_start_y - scale_bar_shift_y,
    
    # Recall that 1 recording time unit = 0.1 ms
    # because it was filtered at 10 kHz
    label = paste0(scale_bar_x_in_recording_time_units/10, " ms"),
    hjust = 0.5,
    family = plot_font_family
  )

ap_trace_insulin <-
  make_AP_trace_plot(file_ID = "23623009_rescaled_0_01",
                     sweep1 = "epi10",
                     sweep2 = "e",
                     trace_color = insulin_group_colour)

dual_plot <- ap_trace_baseline / ap_trace_insulin


if (save_choice == "yes") {
  ggsave(
    plot = dual_plot,
    path = here("Figures", "Output-summary-plots"),
    file = "AP-step-10-representative-traces-AO.png",
    width = 7,
    height = 5,
    units = "in",
    dpi = 300
  )
}
```

```{r AP-data-get-max-frequencies-per-step, include=F}
# --------------- AP Analysis: Get max mean frequencies per step --------------------------------------------------------------------------------------
# Required for correct positioning of significance stars

max_mean_AP_frequencies <- full_ap_df %>%
  group_by(Current_injection, State) %>%
  summarize(
    mean_AP_frequency = mean(AP_frequency),
    SE = sd(AP_frequency) / sqrt(n())
  ) %>%
  group_by(Current_injection) %>%
  summarize(
    max_AP_frequency = max(mean_AP_frequency),
    max_se = max(SE)
  )
```

```{r AP-data-t-test-AP-frequency, include=F}
# --------------------- AP Analysis: T-Test of AP Frequency per Step
# I saved this to a dataframe so I could later add significance asterisks to my plot through geom_text()
# "ns" is distracting in plots, so I replaced it with white-space

# Included here for future testing only
# The data are non-normal, so I used a Wilcoxon test for my analysis

t_test_current_clamp_steps <- full_ap_df %>%
  select(Letter, State, AP_frequency, Current_injection) %>%
  group_by(Current_injection) %>%
  pairwise_t_test(
    AP_frequency ~ State,
    ref.group = "Baseline",
    paired = T,
    p.adjust.method = "holm"
  ) %>%
  mutate(
    statistic = round(statistic, 2),
    p_string = pvalString(p),
    significance_stars = case_when(p.adj.signif == "ns" ~ "",
                             T ~ p.adj.signif)) %>%
  merge(., max_mean_AP_frequencies,
        by = "Current_injection")

t_test_current_clamp_steps %>% 
  select(c(Current_injection, statistic, df, p_string, p.adj.signif)) %>%
  kable(
    col.names = c("Current Injection", "Statistic", "DF", "p-value", ""),
    caption = "Pairwise T-Test with Holm's Correction"
  )
```

```{r AP-data-create-dataset-for-statistical-analysis, include=F}
# Each letter contains 10 rows (one for each sweep)
# This means that the data in columns like 'Threshold', 'Peak_amplitude', 'Time_of_peak' are repeated 10 times
# Create a dataframe for the AP sina plots that does not contain repeated rows

distinct_aps <- full_ap_df %>%
  distinct(Letter, State, .keep_all = TRUE)


# ---------------- AP Analysis: Get differences between control and insulin ------------------------

# For each recording, get the difference in before vs. after values for
# Peak amplitude, latency, etc.
# Use arrange() to ensure that the states are arranged in the proper order within each letter
# These are in the form of
  # (latency_diff) = (latency_insulin) - (latency_baseline)
# Ungroup data and remove NAs to avoid issues later


# Note that there are more values for peak amplitude than the other AP parameters
# This is because some recordings did not have any action potentials
# For these recordings, I wrote "0" for peak amplitude and "NA" for the other AP parameters like half-width

distinct_aps_differences <- distinct_aps %>%
  arrange(Letter, State) %>%
  group_by(Letter) %>%
  mutate(
    Peak_amplitude_diff = Peak_amplitude - lag(Peak_amplitude),
    Latency_diff = Latency_to_fire - lag(Latency_to_fire),
    Threshold_diff = Threshold - lag(Threshold),
    Antipeak_amplitude_diff = Antipeak_amplitude - lag(Antipeak_amplitude),
    Antipeak_time_relative_to_threshold_diff = Antipeak_time_relative_to_threshold - lag(Antipeak_time_relative_to_threshold),
    Half_width_diff = Half_width - lag(Half_width)
  ) %>%
  ungroup() %>%
  drop_na(
    c(
      Peak_amplitude_diff,
      Latency_diff,
      Threshold_diff,
      Antipeak_amplitude_diff,
      Antipeak_time_relative_to_threshold_diff,
      Half_width_diff
    )
  )
```

```{r AP-data-pivot-longer, include=F}
# ---------------------- AP Analysis: Pivot data to longer form -----------------
# Makes it easier to run t-tests across separate AP parameters at once through tidyr framework

wide_ap_df <- distinct_aps %>%
  pivot_longer(
    cols = c(
      "Peak_amplitude",
      "Threshold",
      "Half_width",
      "Latency_to_fire",
      "Antipeak_amplitude",
      "Antipeak_time_relative_to_threshold"
    ),
    names_to = "AP_parameter_name",
    values_to = "AP_parameter_value"
  )

wide_ap_df_differences <- distinct_aps_differences %>%
  pivot_longer(
    cols = c(
      "Peak_amplitude_diff",
      "Threshold_diff",
      "Half_width_diff",
      "Latency_diff",
      "Antipeak_amplitude_diff",
      "Antipeak_time_relative_to_threshold_diff"
    ),
    names_to = "AP_parameter_diff_name",
    values_to = "AP_parameter_diff"
  )
```

```{r AP-data-parameters-t-test, include=F}
# --------------- AP Analysis: T-test for parameters --------------------------------------------------------------------------------------
wide_ap_df %>%
  group_by(AP_parameter_name) %>%
  t_test(
    AP_parameter_value ~ State,
    ref.group = "Baseline",
    paired = T
  ) %>%
  mutate(
    statistic = round(statistic, 2),
    p_original = p,
    p_string = pvalString(p),
    significance_stars = symnum(p, 
                     symbols   = c("***","**","*",""),
                     cutpoints = c(0,  .001,.01,.05, 1),
                     corr      = FALSE
                   )
  ) %>%
  arrange(match(AP_parameter_name, list_of_AP_parameters)) %>%
  select(c(AP_parameter_name, statistic, df, p_string, significance_stars)) %>%
  kable(
    col.names = c("Parameter", "Statistic", "DF", "p-value", ""),
    caption = "Paired T-Test"
  )
```

```{r AP-data-parameters-shapiro-test, include=F}
# --------------- AP Analysis: Testing normality --------------------------------------------------------------------------------------
wide_ap_df_differences %>%
  group_by(AP_parameter_diff_name) %>%
  shapiro_test(AP_parameter_diff) %>%
  select(!variable) %>%
  mutate(
    p_string = pvalString(p),
    p_true = p,
    statistic = round(statistic, 2),
    significance_stars = symnum(
      p,
      symbols   = c("***", "**", "*", ""),
      cutpoints = c(0,  .001, .01, .05, 1),
      corr      = FALSE
    )
  ) %>%
  select(c(
    AP_parameter_diff_name,
    statistic,
    p_string,
    significance_stars
  )) %>%
  arrange(match(
    AP_parameter_diff_name,
    c(
      "Peak_amplitude_diff",
      "Threshold_diff",
      "Half_width_diff",
      "Latency_diff",
      "Antipeak_amplitude_diff",
      "Antipeak_time_relative_to_threshold_diff"
    )
  )) %>%
  kable(col.names = c("Parameter", "Statistic", "P-value", ""),
        caption = "Shapiro-Wilk Normality Test")

```

```{r AP-data-QQ-plots, fig.width = 7, fig.height = 5, include=F, eval=F}
# make_qq_plot(data = distinct_aps_differences,
#              parameter_title = "Peak amplitude",
#              parameter = Peak_amplitude_diff)
# 
# make_qq_plot(data = distinct_aps_differences,
#              parameter_title = "Threshold",
#              parameter = Threshold_diff)
# 
# make_qq_plot(data = distinct_aps_differences,
#              parameter_title = "Half Width",
#              parameter = Half_width_diff)
# 
# make_qq_plot(data = distinct_aps_differences,
#              parameter_title = "Latency",
#              parameter = Latency_diff)
# 
# make_qq_plot(data = distinct_aps_differences,
#              parameter_title = "Antipeak amplitude",
#              parameter = Antipeak_amplitude_diff)
# 
# make_qq_plot(data = distinct_aps_differences,
#              parameter_title = "Antipeak Time",
#              parameter = Antipeak_time_relative_to_threshold_diff)



# The data did not pass normality, and a log-transformation did not work for the data (some of the data were negative, and even when I transformed it, the data were still not normal). Instead, I ran a Wilcoxon signed-rank test.
```

```{r AP-data-wilcoxon-test}
# --------------------- AP Analysis: Wilcoxon Test -------------------------------
wilcoxon_table_ap <- wide_ap_df %>%
  group_by(AP_parameter_name) %>%
  wilcox_test(AP_parameter_value ~ State,
              ref.group = "Baseline",
              paired = T) %>%
  mutate(
    statistic = round(statistic, 2),
    p_string = pvalString(p),
    significance_stars = symnum(
      p,
      symbols   = c("***", "**", "*", ""),
      cutpoints = c(0,  .001, .01, .05, 1),
      corr      = FALSE
    ),
    df = paste(n1, n2, sep = ", "),
    AP_parameter_name = str_replace_all(AP_parameter_name, "_", " "),
    AP_parameter_name = case_when(
      AP_parameter_name == "Half width" ~ "Half-width",
      AP_parameter_name == "Antipeak amplitude" ~ "After-hyperpolarization amplitude",
      AP_parameter_name == "Antipeak time relative to threshold" ~ "After-hyperpolarization time",
      T ~ AP_parameter_name
    )
  ) %>%
  arrange(match(
    AP_parameter_name,
    c(
      "Peak amplitude",
      "Threshold",
      "Latency to fire",
      "Half-width",
      "After-hyperpolarization amplitude",
      "After-hyperpolarization time"
    )
  )) %>%
  select(c(
    AP_parameter_name,
    df,
    statistic,
    p_string
  ))

kable_tidy_wilcoxon <- wilcoxon_table_ap %>%
  kable(booktabs=T,
        linesep = '',
        col.names = c("Parameter", "df", "Statistic", "\\textit{p}-value"),
        escape = F,
        caption = "A Wilcoxon signed-rank test shows that insulin significantly affects action potential amplitudes, thresholds, and half-widths. Insulin does not significantly affect the latency to fire, after-hyperpolarization amplitude, or the time of afterhyperpolarization. \\label{AP-parameters-Wilcoxon-Table}",
        caption.short = "Wilcoxon signed-rank summary table of action potential parameters before and after insulin exposure",
        align = c('l','c','l','l')) %>% 
  kable_styling(full_width = T) %>% 
  column_spec(1,
              width = "2.5in")
```

```{r AP-data-make-summary-ap-frequency-plot, message=F, warning=F}
# ----------------- AP Analysis: Make AP Frequency Plot ---------------------------

full_ap_plot_df <- full_ap_df %>%
  group_by(State, Current_injection) %>%
  summarize(
    mean_AP_frequency = mean(AP_frequency),
    SE = sd(AP_frequency) / sqrt(n()),
    n = n()
  )

ap_plot_baseline_count <- full_ap_plot_df %>% 
  filter(State == "Baseline") %>% 
  pull(n) %>% 
  first()

ap_plot_insulin_count <- full_ap_plot_df %>% 
  filter(State == "Insulin") %>% 
  pull(n) %>% 
  first()

full_ap_frequency_plot <- full_ap_plot_df %>%
  ggplot(
    aes(
      x = Current_injection,
      y = mean_AP_frequency,
      ymin = mean_AP_frequency - SE,
      ymax = mean_AP_frequency + SE,
      color = State,
      shape = State
    )
  ) +
  geom_pointrange(size = 1.1,
                  linewidth = 0.6,
                  position = position_dodge(width = 0)) +
  labs(
    x = "Current Injection (pA)",
    y = "AP Frequency (Hz)",
    color = NULL,
    shape = NULL
  ) +
  scale_color_manual(
    values = c(baseline_group_colour, insulin_group_colour),
    labels = c(
      paste0("Baseline, n = ", ap_plot_baseline_count),
      paste0("Insulin, n = ", ap_plot_insulin_count)
    )
  ) +
  scale_shape_manual(
    values = c(16, 17),
    labels = c(
      paste0("Baseline, n = ", ap_plot_baseline_count),
      paste0("Insulin, n = ", ap_plot_insulin_count)
    )
  ) +
  theme(legend.position = "inside",
        legend.position.inside = c(0.18, 0.8),
        legend.key.spacing.y = unit(1.5, "lines"),
        axis.title = element_text(face = "plain")) +
  scale_y_continuous(expand = expansion(mult = c(0.1, .1))) +
  geom_text(
      data = t_test_current_clamp_steps,
      aes(
        x = Current_injection,
        y = max_AP_frequency + max_se + 1,
        label = significance_stars
      ),
      inherit.aes = F,
      size = 5,
      family = significance_stars_font
    )
```

```{r AP-data-make-raw-ap-frequency-plot}
raw_ap_frequency_plot <- full_ap_df %>%
  ggplot(aes(x = Current_injection, y = AP_frequency, color = State)) +
  geom_point(position = "jitter") +
  labs(x = "Current Injection (pA)",
       y = "AP Frequency (Hz)",
       color = NULL) +
  scale_color_manual(values = c(baseline_group_colour, insulin_group_colour)) +
  theme(legend.position = "inside",
        legend.position.inside = c(0.11, 0.8),
        axis.title = element_text(face = "plain")) +
  scale_y_continuous(expand = expansion(mult = c(0.1, .1)))
```

```{r AP-data-make-AP-characteristics-plots}
# ------------------ AP Analysis: Make AP Characteristics Plots ---------------------
peak_amplitude_plot <- make_AP_plot(data = distinct_aps,
                                    y = Peak_amplitude,
                                    y_axis_title = "Amplitude (mV)")

latency_plot <- make_AP_plot(data = distinct_aps,
                             y = Latency_to_fire,
                             y_axis_title = "Latency (ms)")

threshold_plot <- make_AP_plot(data = distinct_aps,
                               y = Threshold,
                               y_axis_title = "Threshold (mV)")

antipeak_amplitude_plot <- make_AP_plot(data = distinct_aps,
                                        y = Antipeak_amplitude,
                                        y_axis_title = "Afterhyperpolarization\nAmplitude (mV)")

time_of_antipeak_plot <- make_AP_plot(
  data = distinct_aps,
  y = Antipeak_time_relative_to_threshold,
  y_axis_title = "Time of After-\nhyperpolarization (ms)"
)

half_width_plot <- make_AP_plot(data = distinct_aps,
                                y = Half_width,
                                y_axis_title = "Half Width (ms)")


# Save Plots
if (save_choice == "yes") {
  save_AP_plot(full_ap_frequency_plot, "AP-frequency")
  save_AP_plot(peak_amplitude_plot, "Peak-amplitude")
  save_AP_plot(latency_plot, "Latency")
  save_AP_plot(threshold_plot, "Threshold")
  save_AP_plot(half_width_plot, "Half-width")
  save_AP_plot(antipeak_amplitude_plot, "Antipeak-amplitude")
  save_AP_plot(time_of_antipeak_plot, "Time-of-Antipeak")
}
```

```{r eEPSC-data-import-data, include = FALSE}
# --------------- eEPSC Analysis --------------------------------------------------------------------------------------

raw_df <- read.csv(here("Data/Raw-eEPSC-Data.csv"), header = T) %>%
  
  # Remove letters with known issues
  filter(!Letter %in% c("GE", "GN", "GJ"))


# Evoked current amplitudes were negative in Clampex
# Negative transform P1 and P2 for easier interpretation
# Add paired pulse ratio (PPR column)

# Must define experimental categories as factors to reduce errors (R thinks these are numeric)
# Time filtering (Time <= 25) is required for consistency in plots


raw_df <- raw_df %>%
  mutate(P1 = P1 * -1,
         P2 = P2 * -1,
         PPR = P2 / P1,
         Category = factor(Category)) %>%
  filter(Time <= 25)


# Must define experimental categories as factors to reduce errors (R thinks these are numeric)



raw_df$Category <- factor(raw_df$Category)


# --------------- eEPSC Analysis: Normalize currents relative to baseline ---------------------------------------------------------

# The first part of the code below creates time intervals of 0-5, 6-10, 11-15, etc.
  # I included -1 in the list of breaks so that t = 0 would be included
  # Add a new column with the names for each interval.
  # I tried naming the intervals '0-5', but R thought I was trying to subtract and it did not recognize the names as characters. Instead, I used 't0to5'.

# The second part of this code normalizes each current as a percentage relative to its own baseline
  # Group by letter

  # The baseline is the mean P1 of the first 5 minutes
  # Used time <= 5 to include only the first 5 minutes in the baseline period
  # Get the mean current amplitude for the baseline (0-5 min) for each cell

  # Create a new variable called P1_transformed
  # Express each P1 relative as a percentage relative to its own baseline, which is ~100%
  # This normalizes the P1 to the baseline
  # Repeat this process for P2

raw_df <- raw_df %>%
  mutate(Interval = cut(
    Time,
    breaks = c(-1, 5, 10, 15, 20, 25),
    labels = c("t0to5", "t5to10", "t10to15", "t15to20", "t20to25")
  )) %>%
  group_by(Letter) %>%
  mutate(
    baseline_range = (Time <= 5),
    baseline_mean = sum(P1 * baseline_range) / sum(baseline_range),
    P1_transformed = (P1 / baseline_mean) * 100,
    P2_transformed = (P2 / baseline_mean) * 100,
    Animal_factor = as.factor(unique(Animal))
  )

# Write the raw data to a RDS file for use in other documents/analyses
saveRDS(raw_df, file = here("Data/Output-Data-from-R/raw_df.RDS"))

# -------------- eEPSC Analysis: Make dataset for statistical testing -------------------------------------------------------
# Group by letter, sex, treatment and interval
# Get mean P1_transformed amplitude and standard error for each 5-min time interval
# Unfortunately, reframe() removes columns that do not have a function applied to them.
# Use unique() to retain the columns for age, animal, location for each cell and experimental category

summary_df <-
  raw_df %>%
  group_by(Letter, Sex, Treatment, Interval) %>%
  reframe(
    Mean = mean(P1_transformed),
    SE = sd(P1_transformed) / sqrt(n()),
    Age = unique(Age),
    Animal = unique(Animal),
    Animal_factor = as.factor(unique(Animal)),
    n = n(),
    X = unique(X),
    Y = unique(Y),
    Category = unique(Category)
  )
```

```{r eEPSC-data-make-raw-plots, include = FALSE}
# ----------------- eEPSC Analysis: Plot P1 vs. Time --------------------------------------------------------------
# Use custom function make_raw_plots().
# 1st argument: Raw dataframe
# 2nd argument: The experimental category
# Recall from above if you set "save_choice" to yes or no

# This can take a long time to run, and I don't need to run it each time I run the thesis
# Uncomment the line below if you want to produce the raw plots

# make_raw_plots(dataframe = raw_df,
#                treatment = "Control",
#                category = 2,
#                color_choice = 1)


# ----------------- eEPSC Analysis: View raw plots -----------------------------------------------------------------------------
# Not for publication; just for quick viewing in R
# Written in the form of Individual_raw_plots_[category number]
# This number should match argument 2 of make_raw_plots(), or you will see an "object not found" error

# These plots are super useful for seeing individual recordings, but I do not want them in the thesis
# It also can take a while to run this line
# Uncomment the line below if you want to see the raw plots

#Individual_raw_plots_Fasting_category_2
```

```{r eEPSC-data-individual-pruned-df, include=FALSE}
# ----------------- eEPSC Analysis: Prune dataset per minute ------------------------------------------------------------------
# GraphPad Prism has a "prune rows" command that outputs the mean and standard error of every n rows.
# I performed an equivalent pruning function in R.
# I used by = 1 because it will generate one summary point per minute.

# Group raw data by category
# Within each category, define intervals of 1-min each
# Used include.lowest = T so that t = 0 would not be deleted
# 1st group_by(): Get the mean amplitude for each minute within each cell
# 2nd group_by(): Get the mean amplitude for each minute across ALL cells
# Keep baseline_mean because this is required to create horizontal annotation of the baseline

pruned_df_individual_cells <- raw_df %>%
  group_by(Category) %>%
  mutate(Interval_pruned = cut(
    Time,
    breaks = seq(0, 25, by = 1.5),
    include.lowest = T
  )) %>%
  group_by(Category, Letter, Sex, Treatment, Interval_pruned) %>%
  reframe(
    mean_P1 = mean(P1),
    sd_P1 = sd(P1),
    n = n(),
    se = sd_P1 / sqrt(n),
    Letter = unique(Letter),
    Category = unique(Category),
    Time = last(Time),
    baseline_mean = unique(baseline_mean)
  )


# Write the pruned data to a RDS file for use in other documents/analyses
saveRDS(pruned_df_individual_cells, file = here("Data/Output-Data-from-R/pruned_df_individual_cells.RDS"))
```

```{r eEPSC-data-make-individual-pruned-plots}
# ----------------- eEPSC Analysis: Plot pruned P1 vs. Time --------------------------------------------------------------
# If save_choice = "yes", the plots will be saved in the form of '[Letter]-pruned.png'
# Use custom function make_individual_pruned_plots().
# 1st argument: Pruned individual cells dataframe
# 2nd argument: The experimental category

# make_individual_pruned_plots(data = pruned_df_individual_cells,
#                              treatment = "Control",
#                              category = 2,
#                              color_choice = 1)


# -------------- eEPSC Analysis: View pruned plots ---------------------------------------------------------
# These plots are not part of the thesis, but they are helpful to monitor cells

#Individual_pruned_plots_Control_category_1

```

```{r eEPSC-data-make-pruned-df-full}
# ---------------- eEPSC Analysis: Make pruned dataframe for multiple cells ------------------------------------------------------
# Within each category, define intervals of 1-min
# First group_by: Get the mean amplitude for each minute within each cell
# Second group_by: Get the mean amplitude for each minute across ALL cells 
pruned_df <- raw_df %>%
  group_by(Category) %>%
  mutate(Interval_pruned = cut(
    Time,
    breaks = seq(0, 25, by = 1),
    include.lowest = T
  )) %>%
  group_by(Category, Letter, Sex, Treatment, Interval_pruned) %>%
  reframe(
    mean_P1 = mean(P1_transformed),
    sd_P1 = sd(P1_transformed),
    n = n(),
    se = sd_P1 / sqrt(n),
    Letter = unique(Letter),
    Category = unique(Category),
    Time = last(Time)
  ) %>%
  group_by(Category, Interval_pruned, Sex, Treatment) %>%
  reframe(
    Mean_all = mean(mean_P1),
    sd_all = sd(mean_P1),
    n = n(),
    se = sd_all / sqrt(n),
    Time = last(Time),
    Category = unique(Category)
  )

# Write the pruned data to a RDS file for use in other documents/analyses
saveRDS(pruned_df, file = here("Data/Output-Data-from-R/pruned_df.RDS"))
```

```{r eEPSC-data-create-wide-format, purl = FALSE}
# ----------------- eEPSC analysis: Begin statistical analysis ---------------------------------

# The wide version of the dataframe is required for the MANOVA model
summary_df_wide <- summary_df %>%
  select(!c(SE, n)) %>%
  pivot_wider(
    names_from = Interval,
    values_from = Mean
  )
```

```{r eEPSC-data-create-insulin-only-df, purl = FALSE}
# ---------------- eEPSC Analysis: Filter only complete datasets ----------------------------------------------------------
# Complete datasets are where I added insulin (Category == 2) for no treatment, HNMPA, PPP, or Fasting.
# The other treatments are incomplete, so they are not ready for statistical analysis

summary_df_insulin <- summary_df %>%
  filter(Category == 2) %>%
  filter(Treatment %in% list_of_treatments)

# Do the same filtering as above, but on the wide dataset
summary_df_wide_addinsulin <- summary_df_wide %>%
  filter(Category == 2) %>%
  filter(Treatment %in% list_of_treatments)
```

```{r eEPSC-data-MANOVAR-adding-insulin, include=FALSE, purl = FALSE}
# This dataframe of times is required to generate the linear model
Times <-
  factor(rep(
    c("t0to5", "t5to10", "t10to15", "t15to20", "t20to25"),
    c(1, 1, 1, 1, 1)
  ))


# Force desired level order instead of alphabetical orders
Times <-
  factor(Times,
    levels = c("t0to5", "t5to10", "t10to15", "t15to20", "t20to25")
  )

idata <- data.frame(Times)


# -------------- eEPSC Analysis: Create a linear model --------------------------------------------------------------------------
# Models time as a function of treatment and sex
# Used treatment*sex to account for treatment, sex, and the interaction between treatment and sex

model.fit <- lm(cbind(t0to5, t5to10, t10to15, t15to20, t20to25) ~ Treatment*Sex, data = summary_df_wide_addinsulin)


# Generate a repeated measures model using the times created above
rep_measures_insulin <- Anova(model.fit, idata = idata, idesign = ~Times)

# Look at this to get Pillai's trace values
# See MANOVAR summary table below for reader-friendly view of this
#rep_measures_insulin

# View the model output; hidden because it is too long to display
# See MANOVAR summary table code below for a reader-friendly view
summary_manovar <- summary(rep_measures_insulin)
#summary_manovar
```

```{r eEPSC-data-get-manovar-dataframe, warning=F, message=F}
# -------------------------- eEPSC Analysis: Get MANOVAR summary table as object -------------------
# print.Anova.mlm prints the dataframe but uses an invisible object
# This meant that I could see the dataframe, but it was not available as an object
# I could not extract the p-values from a class Anova.mlm object, and manually typing them was not ideal for reproducibility

# This answer shows how to extract the summary table as a dataframe from an Anova.mlm object
# https://stackoverflow.com/questions/65161615/extracting-p-value-column-from-output-anova-car-package
# The following function will remove the last two lines of the print.Anova.mlm function in the car package that would normally use an invisible function to delete the dataframe
# Using this modified function will return a dataframe, where you can extract the p-values directly

get_summary_for_print <-
  car:::print.Anova.mlm # copy the original print function (inclusive environment)
body(get_summary_for_print) <- # replace the code of our copy
  local({
    # to avoid pollution of environment by tmp
    tmp <-
      body(get_summary_for_print) # to avoid code duplication
    tmp[-(length(tmp) - (0:1))] # remove the last two code lines of the function
  })

# Use get_summary_for_print() to extract the dataframe
# Use tidy() to clean up the column names and dataframe structure
# Create reader-friendly DF column, rounded test statistics and F-values, and clearer column names
# Use pvalString to express small p-values as <0.0001 instead of exact values.

tidy_manovar <- get_summary_for_print(rep_measures_insulin) %>%
  tidy() %>%
  mutate(df = paste(num.Df, den.Df, sep = ", ")) %>%
  select(!c(num.Df, den.Df)) %>%
  mutate(across(c("test.stat", "approx.F"), \(x) round(x, 2)),
         p.value = pvalString(p.value)) %>%
  rename(
    Term = term,
    'Pillai\'s Trace' = test.stat,
    F = approx.F,
    'p-value' = p.value
  )
```

```{r eEPSC-get-publication-ready-manovar-table}
# -------------------------- eEPSC Analysis: Make publication-ready MANOVAR summary table -------------------
# Use booktabs package to get publication-quality table in LaTeX PDF output
kable_manovar_table_eEPSC <- tidy_manovar %>%
  kable(booktabs = T,
        linesep = '',
        col.names = c("Parameters","df","Pillai's Trace","F","\\textit{p}-value"),
        escape = F,
        caption = "A multivariate repeated measures ANOVA using Pillai's Trace shows that current amplitude varies significantly with time, and there are no significant interactions between factors like sex and treatment. \\label{MANOVAR-Table}",
        caption.short = "MANOVAR summary table of evoked excitatory currents before and during insulin exposure",
        align = c('l','r','r','r','r')) %>%
  kable_styling(full_width = T) %>%
  column_spec(1,
              width = "1.7in")
```

```{r eEPSC-data-show-sphericity-output, include = FALSE, purl = FALSE}
# ----------------- eEPSC Analysis: Assumption Test - sphericity ----------------------------------------------------------
summary_manovar$sphericity.tests


# Sun Mar 24 17:29:52 2024
# The p-values for Mauchly's sphericity tests were all much lower than 0.05.
# This means that my data failed Mauchly's test of sphericity.
# I switched to a multivariate repeated measures model, which does not require the assumption of sphericity.
```

```{r eEPSC-data-univariate-levene-s-test, include = FALSE, purl = FALSE}
# ---------------- eEPSC Analysis: Assumption test - univariate homogeneity of variances --------------------------------
summary_df_insulin %>%
  group_by(Interval) %>%
  mutate(Treatment = factor(Treatment)) %>%
  levene_test(Mean ~ Treatment) %>%
  mutate(statistic = round(statistic, 3),
         p = round(p, 3))

# Sun Mar 24 17:30:36 2024
# The data all passed Levene's Test, indicating that all of the data passed the univariate tests for the homogeneity of variances.
```

```{r eEPSC-data-univariate-model, include = FALSE, purl = FALSE}
# ---------------- eEPSC Analysis: Create univariate model for assumption testing -------------------
# Filter to remove baseline data, since all values are 100 and there is no variance.
# Otherwise, this will generate error messages
summary_df_insulin_no_baseline <- summary_df_insulin %>%
  filter(Interval != "t0to5")

univariate_model <-
  lme(
    Mean ~ Treatment * Interval * Sex,
    data = summary_df_insulin_no_baseline,
    random = ~ 1 |
      Sex / Interval,
    correlation = corCompSymm(form = ~ 1 | Sex / Interval)
  )
anova(univariate_model)
```

```{r eEPSC-data-univariate-normality, include = FALSE, purl = FALSE}
# ---------------- eEPSC Analysis: Assumption test - Univariate normality ------------------------------------------------------
univariate_model_resid <- resid(univariate_model)
shapiro.test(univariate_model_resid)

qqnorm(univariate_model_resid)
qqline(univariate_model_resid)


# Sun Mar 24 17:35:09 2024
# The p-values for the Shapiro-Wilk test are 0.02, indicating that the data did not pass the univariate assumptions of normality.
# The QQ-plot looks better than in previous versions, however, since the points are more linear
```

```{r eEPSC-data-multivariate-box-s-M, include = FALSE, purl = FALSE}
# ---------------- eEPSC Analysis: Assumption test - Multivariate homogeneity of variances -------------------------------------
# I excluded the baseline (t0to5) because all values were 100% - otherwise this will generate errors
boxM(cbind(t5to10, t10to15, t15to20, t20to25) ~ Treatment,
  data = summary_df_wide_addinsulin
)


# Sun Mar 24 17:34:03 2024
# The p-value for Box's M is 0.2933, indicating that the covariance matrices are homogeneous.
```

```{r eEPSC-data-multi-shapiro, include = FALSE, purl = FALSE}
# ---------------- eEPSC Analysis: Assumption test - Multivariate normality ----------------------------------------------------
# Note: I used a test range of 5 to 25 minutes
# This is because the values in the 0-5 minute range (column 10) were all 100%.
norm_test_range <- summary_df_wide_addinsulin %>%
  select('t5to10':'t20to25')

# ! Data are from 0 to 25 min
mshapiro_result <- mshapiro.test(norm_test_range)


# ! Data are from 0 to 25 min
qqnorm<-mqqnorm(norm_test_range)


# Sun Mar 24 17:36:37 2024
# The p-value for the multivariate Shapiro-Wilk normality test was 0.004066, indicating that the data failed the assumption of normality.
  
# However, the points are within the 95% confidence interval on the Multi-QQ plot.

# Sun Mar 24 17:37:31 2024
# Although the data failed both the univariate and multivariate Shapiro-Wilk tests, I felt confident in assuming normality after visually assessing the QQ plots.

# Reasons:
# 1: The points in the univariate QQ-plot were roughly linear

# 2: The points within the multivariate QQ-plot were contained within the 95% confidence interval
```

```{r eEPSC-data-t-tests, include=F, purl = F}
# ---------------- eEPSC Analysis: T-tests over time ---------------------------------------------------------------------------------------
# I used paired = T because the data were taken within the same cells (25 min later)
# Paired=T requires vectors of equal length
# I could not use paired t-tests on the full dataset because there were missing values in the 25-30 min interval.
# This is why I filtered all data to contain Time values of 25 or less.


# I rounded the p-values and t-statistic for better readability.
# I added a column called asterisk_time, which codes for the x-position of the significance asterisks in my plots.
# asterisk_time uses the upper time limit of each interval
# For example, it will place the significance asterisk for interval t0to5 at time = 5

# I used case_when() to change "ns" to a blank space ""
# This is because having "ns" on the plots was visually distracting

t_test_eEPSCs <- summary_df_insulin %>%
  group_by(Treatment) %>%
  pairwise_t_test(
    Mean ~ Interval,
    ref.group = "t0to5",
    paired = T,
    p.adjust.method = "holm"
  ) %>%
  mutate(statistic = round(statistic, 2),
         p_string = pvalString(p.adj),
         asterisk_time = as.numeric(case_when(
           group2 == "t5to10" ~ "7.5",
           group2 == "t10to15" ~ "12.5",
           group2 == "t15to20" ~ "17.5",
           group2 == "t20to25" ~ "22.5",
           T ~ "Text"
         )),
         significance_stars = case_when(
           p.adj.signif == "ns" ~ "",
           T ~ p.adj.signif
         ))

saveRDS(t_test_eEPSCs, file = here("Data/Output-Data-from-R/t_test_eEPSCs.RDS"))
```

```{r PPR-data-create-dataframe-for-t-test, warning=F, message=F}
# -------------------- PPR Analysis: Create dataframe for t-tests ----------------
PPR_df <- raw_df %>%
  filter(Category == 2) %>%
  filter(Treatment %in% list_of_treatments) %>%
  filter(PPR < 5 & PPR > 0) %>%
  filter(Interval == "t0to5" | Interval == "t20to25") %>%
  mutate(
    Treatment = factor(Treatment, levels = list_of_treatments),
    State = case_when(
      Interval == "t0to5" ~ "Baseline",
      Interval == "t20to25" ~ "Insulin",
      T ~ Interval
    )
  )
```

```{r PPR-data-t-test, warning=F, message=F}
t_test_PPR <- PPR_df %>%
  group_by(Treatment, Letter, State) %>%
  summarise(mean_PPR_cell = mean(PPR),
            n = n()) %>%
  ungroup() %>%
  group_by(Treatment) %>%
  t_test(mean_PPR_cell ~ State,
         ref.group = "Baseline",
         paired = T) %>%
  mutate(
    statistic = round(statistic, 2),
    p_original = p,
    p_string = pvalString(p),
    significance_stars = symnum(
      p,
      symbols   = c("***", "**", "*", "ns"),
      cutpoints = c(0,  .001, .01, .05, 1),
      corr      = FALSE
    )
  ) %>%
  arrange(match(Treatment, list_of_treatments))


kable_t_test_table_PPR <- t_test_PPR %>%
  select(c(Treatment, statistic, df, p_string)) %>%
  kable(
    booktabs = T,
    linesep = '',
    col.names = c("Parameter", "Statistic", "DF", "\\textit{p}-value"),
    caption = "A paired t-test comparing the mean paired pulse ratio (PPR) per cell during the baseline period (0-5 min) to the mean PPR after insulin exposure (20-25 min) shows that insulin does not significantly affect the PPR during experiments with no additional treatments, fasting, or PPP. Insulin significantly increases the PPR when the insulin receptor blocker HNMPA is applied. \\label{PPR-paired-t-test}",
    caption.short = "Paired T-test summary table examining paired pulse ratios before and after insulin exposure",
    escape = F
  ) %>%
  kable_styling(full_width = T) %>%
  column_spec(1, width = "2.5in")
```

```{r PPR-data-create-differences-df, message=F}
# -------------- PPR Analysis: Get differences between control and insulin ------------------------
# Create a dataframe containing the differences in PPR from insulin to baseline
# arrange() sorts the data in the form of baseline -> insulin within each letter
# This enables me to use lag() to get:
# PPR_diff = PPR_insulin - PPR_baseline

PPR_differences_df <- PPR_df %>%
  group_by(Treatment, Letter, State) %>%
  summarise(mean_PPR_cell = mean(PPR),
            n = n()) %>%
  ungroup() %>% 
  arrange(Letter, State) %>%
  group_by(Letter) %>%
  mutate(
    PPR_diff = mean_PPR_cell - lag(mean_PPR_cell)
    ) %>%
  ungroup() %>%
  drop_na(PPR_diff)

```

```{r PPR-data-shapiro, message=F, warning=F, include = F}
# ----------------PPR Analysis: Assumption test - Normality
shapiro.test(PPR_differences_df$PPR_diff)

# Sat Mar 30 17:45:29 2024
# The data are normal, indicating that I can use the results from the t-test
```

```{r cell-coordinates-df}
# ---------------- Cell Locations: Modify dataset for cell locations plot ------------------------------------------------------------------
# Filter out cells with missing coordinates
# Add a new column called percent change that looks at the percent change 20 minutes after adding insulin
# Filter (for now) to only include full datasets for the experiments where I added insulin (Category == 2)

cell_coordinates <-
  summary_df %>%
  group_by(Treatment, Interval, Letter) %>%
  select(!c(SE,n)) %>%
  pivot_wider(
    names_from = Interval,
    values_from = Mean
  ) %>%
  mutate(
    percent_change = t20to25/t0to5*100
  ) %>%
  filter_at(vars(X, Y), all_vars(!is.na(.))) %>% 
  filter(Category == 2) %>%
  filter(Treatment %in% list_of_treatments)

saveRDS(cell_coordinates, file = here("Data/Output-Data-from-R/cell_coordinates_with_percent_change.RDS"))

background_slice_image <- jpeg::readJPEG(here("Figures/DMH-Image.jpg"))
```

```{r eEPSC-get-maximum-percent-change}
# --------------------- eEPSC Analysis: Get max percent change after 20 min -----------------------

# Get the value of the maximum percent change after 20 minutes
# I filtered this to only include the treatments of interest
# This is required to set the scale in the cell locations plot.
# plyr::round_any rounds this value up to the nearest multiple of 10
# Required to create 'nice' scale limits
max_percent_change <- cell_coordinates %>%
  ungroup() %>%
  summarise(
    max(percent_change)) %>%
  pull() %>%
  plyr::round_any(10, ceiling)
```

\cleardoublepage

# Results

The primary goal of this study was to examine the effect of insulin on neuronal excitability and communication in DMH neurons. I used whole-cell patch clamp electrophysiology to record action potentials and evoked excitatory currents from `r nrow(summary_df_wide_addinsulin)` unique neurons in the DMH. I analyzed recordings taken before, while and after DMH neurons were exposed to 500 nM insulin. By comparing parameters like action potential frequency and evoked current amplitude across conditions, I was able to draw conclusions about the effect of insulin on intrinsic neuronal excitability, excitatory synaptic transmission, and neurotransmitter release onto DMH neurons.

The second goal of the study was to determine the mechanism of insulin signalling within DMH neurons. I used receptor blockers like HNMPA (which blocks insulin receptors) and PPP (which blocks insulin-like growth factor 1 receptors) to determine the receptors that insulin binds to, and to explore intracellular signalling pathways that might be activated by insulin binding. I also examined changes in the paired pulse ratio to determine if insulin signalling uses pre- or post-synaptic mechanisms. Lastly, I exposed rats to a 24-hour fasting protocol before tissue extraction to assess if hunger affects insulin signalling within the DMH relative to control experiments from satiated animals.

## DMH neurons are less excitable with insulin

A current clamp steps protocol taken before and after insulin exposure showed that insulin decreases action potential frequency (Figure \ref{ap-multiplot-figure}A). The difference in action potential frequency was statistically significant from a current injection of -10 pA (\emph{t}(`r t_test_current_clamp_steps %>%   filter(Current_injection == -10) %>%   pull(df)`) = `r t_test_current_clamp_steps %>%   filter(Current_injection == -10) %>%   pull(statistic)`, \emph{p} = `r t_test_current_clamp_steps %>%   filter(Current_injection == -10) %>%   pull(p_string)`) up to the highest current injection of 40 pA (\emph{t}(`r t_test_current_clamp_steps %>%   filter(Current_injection == 40) %>%   pull(df)`) = `r t_test_current_clamp_steps %>%   filter(Current_injection == 40) %>%   pull(statistic)`, \emph{p} = `r t_test_current_clamp_steps %>%   filter(Current_injection == 40) %>%   pull(p_string)`).

```{r AP-data-multiplot-figure, fig.width = 10, fig.height = 12, warning=F}
#| fig.cap = "Insulin decreases the excitability of DMH neurons. Data were recorded before and then after 25 minutes of insulin exposure. \\emph{A)} Insulin significantly decreases action potential frequency (mean ± SE; \\emph{n} is the number of cells). \\emph{B)} Representative traces from a current injection of 40 pA (top: baseline, bottom: insulin). \\emph{C)} Insulin significantly decreases action potential amplitudes \\emph{C)} and thresholds \\emph{D)}, while significantly increasing half-widths \\emph{E)}. Insulin does not significantly affect latency to fire \\emph{F)}, after-hyperpolarization amplitude \\emph{G)} or after-hyperpolarization time \\emph{H)}. Overlay on Figures C-H: mean ± SE. \\emph{n} = 15 for baseline and \\emph{n} = 11 for insulin because 4 cells did not fire any action potentials after insulin exposure. Wilcoxon signed-rank test, * = p < 0.05, ** = p < 0.01, *** = p < 0.001. \\label{ap-multiplot-figure}",
#| fig.scap = "Insulin changes action potential characteristics in DMH neurons",
#| out.width = "97%"

# ----------------------- AP Analysis: Full Multiplot Figure -----------------------------------
full_ap_multiplot <-
  (full_ap_frequency_plot |
     (ap_trace_baseline / ap_trace_insulin)) / (peak_amplitude_plot + threshold_plot + half_width_plot) /
  (latency_plot + antipeak_amplitude_plot + time_of_antipeak_plot) +
  plot_layout(heights = c(0.8, 0.5, 0.5)) &
  theme(
    axis.title = element_text(size = 12, face = "plain"),
    axis.title.y = element_text(margin = margin(r = 10)),
    axis.text = element_text(size = 8),
    plot.margin = margin(10, 20, 10, 5)
  )


# Apply theme_void to just the representative traces plots since one already has a scale bar and both of them have been set to the same y-axes

full_ap_multiplot[[1]][[2]][[1]] <- full_ap_multiplot[[1]][[2]][[1]] + theme_void()
full_ap_multiplot[[1]][[2]][[2]] <- full_ap_multiplot[[1]][[2]][[2]] + theme_void()


# Manually applied a list of labels because I wanted the second trace to be blank
# baseline and insulin sweeps both count as figure "B"
# Styled tags separately since theme_void on plots "B" and "[c]" changed the style

full_ap_multiplot <- full_ap_multiplot +
  plot_annotation(tag_levels = list(c('A','B','','C','D','E','F','G','H'))) &
  theme(
    plot.tag = element_text(
      size = 18,
      margin = margin(r = 10, b = 10),
      face = "bold"
    )
  )

full_ap_multiplot
```

Insulin significantly decreases action potential amplitude relative to baseline amplitudes ($W$(`r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Peak amplitude") %>%   pull(df)`) = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Peak amplitude") %>%   pull(statistic)`, \emph{p} = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Peak amplitude") %>%   pull(p_string)`; Figure \ref{ap-multiplot-figure}C). In four recordings, there were no amplitudes after insulin exposure, so additional action potential parameters could not be measured. Of the action potentials that did fire after insulin exposure, the threshold decreased significantly ($W$(`r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Threshold") %>%   pull(df)`) = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Threshold") %>%   pull(statistic)`, \emph{p} = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Threshold") %>%   pull(p_string)`; Figure \ref{ap-multiplot-figure}D). Action potential half-widths increased significantly with insulin ($W$(`r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Half-width") %>%   pull(df)`) = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Half-width") %>%   pull(statistic)`, \emph{p} = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Half-width") %>%   pull(p_string)`; Figure \ref{ap-multiplot-figure}E). The latency to fire did not change significantly after insulin exposure ($W$(`r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Latency to fire") %>%   pull(df)`) = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Latency to fire") %>%   pull(statistic)`, \emph{p} = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "Latency to fire") %>%   pull(p_string)`; Figure \ref{ap-multiplot-figure}F). The amplitude ($W$(`r wilcoxon_table_ap %>%   filter(AP_parameter_name == "After-hyperpolarization amplitude") %>%   pull(df)`) = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "After-hyperpolarization amplitude") %>%   pull(statistic)`, \emph{p} = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "After-hyperpolarization amplitude") %>%   pull(p_string)`; Figure \ref{ap-multiplot-figure}G) and time of the after-hyperpolarization ($W$(`r wilcoxon_table_ap %>%   filter(AP_parameter_name == "After-hyperpolarization time") %>%   pull(df)`) = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "After-hyperpolarization time") %>%   pull(statistic)`, \emph{p} = `r wilcoxon_table_ap %>%   filter(AP_parameter_name == "After-hyperpolarization time") %>%   pull(p_string)`; Figure \ref{ap-multiplot-figure}H) also did not significantly change after insulin exposure.

```{r}
kable_tidy_wilcoxon
```

## Excitatory synaptic transmission decreases after insulin exposure

To assess the effect of insulin on excitatory synaptic transmission in DMH neurons, I compared eEPSC amplitudes with insulin relative to a baseline period under control conditions. Insulin significantly decreased current amplitude over time in cells with no additional treatments applied (\emph{p} `r tidy_manovar$'p-value'[5]`; Figure \ref{summary-plot-control}). The decrease in evoked current amplitudes was statistically significant within the first five minutes of adding insulin (\emph{t}(`r t_test_eEPSCs %>%   filter(Treatment == "Control" & group2 == "t5to10") %>%   pull(df)`) = `r t_test_eEPSCs %>%   filter(Treatment == "Control" & group2 == "t5to10") %>%   pull(statistic)`, \emph{p} = `r t_test_eEPSCs %>%   filter(Treatment == "Control" & group2 == "t5to10") %>%   pull(p_string)`). This indicates that insulin binding quickly activates intracellular processes that decrease excitatory synaptic transmission in the DMH.

The changes in evoked current amplitudes were not significantly different across sexes (Sex\*Times, \emph{p} = `r tidy_manovar$'p-value'[7]`), suggesting that insulin acts independently of sex in the DMH. I also found no cross-factor interactions between sex and treatment (Treatment\*Sex\*Times, \emph{p} = `r tidy_manovar$'p-value'[8]`).

```{r import-representative-traces}
# Import representative traces
male_control_trace <- png::readPNG(here("Figures/Representative-Traces/Male-Control-Trace-H.png")) %>% rasterGrob()
female_control_trace <- png::readPNG(here("Figures/Representative-Traces/Female-Control-Trace-AV.png")) %>% rasterGrob()
male_HNMPA_trace <- png::readPNG(here("Figures/Representative-Traces/Male-HNMPA-Trace-BO.png")) %>% rasterGrob()

male_fasting_trace <- png::readPNG(here("Figures/Representative-Traces/Male-Fasting-Trace-FG.png")) %>% rasterGrob()

female_PPP_trace <- png::readPNG(here("Figures/Representative-Traces/Female-PPP-Trace-GF.png")) %>% rasterGrob()


```

```{r plot-control}
#| fig.dim = c(8,5),
#| fig.cap = "Insulin significantly decreases excitatory evoked post-synaptic current (eEPSC) amplitude over time in both sexes, with no significant differences between the sexes. I exposed DMH neurons to 500 nM of insulin from 5 minutes and onward. Each point represents the mean eEPSC amplitude (± the standard error) across all cells and \\emph{n} represents the number of unique cells. The asterisks indicate a statistically significant decrease in current amplitude relative to the baseline (t-test; * = p < 0.05, ** = p < 0.01, *** = p < 0.001). The representative traces consist of eEPSCs from one cell averaged over the baseline period (0 to 5 min) and last interval (20 to 25 min) of the recording, and the scale bar represents 50 pA/20 ms. \\label{summary-plot-control}",
#| fig.scap = "Insulin significantly decreases evoked excitatory current amplitudes in DMH neurons",
#| out.width = "100%"


# ---------------- Get summary plots for all cells --------------------------------------------------------------
# Uses custom function make_pruned_summary_plots()
# 1st argument: Experiment category
# 2nd argument: Treatment
# 3rd argument: pruned dataframe
# 4th argument: Choose colour from colour palette lists defined at the top of the script
# e.g. 1 corresponds to the 1st item in the list, which is purple
make_pruned_summary_plots(category = 2,
                         treatment = "Control",
                         data = pruned_df,
                         color_choice = 1,
                         representative_trace = female_control_trace)

```

```{r}
kable_manovar_table_eEPSC
```

## Insulin may bind to insulin-like growth factor 1 receptors on DMH neurons

After verifying that insulin acts on DMH neurons, the next step was to determine which receptors insulin can bind to. I performed the same evoked currents protocol, but with HNMPA in the intracellular solution. HNMPA is an insulin receptor antagonist that inhibits tyrosine kinase activity [@sapersteinetal1989]. Since insulin receptors belong to the tyrosine kinase superfamily [@demeyts2000], HNMPA would have blocked insulin from binding to insulin receptors. However, insulin also significantly decreased evoked current amplitudes within the first 5 minutes of exposure (\emph{t}(`r t_test_eEPSCs %>%   filter(Treatment == "HNMPA" & group2 == "t5to10") %>%   pull(df)`) = `r t_test_eEPSCs %>%   filter(Treatment == "HNMPA" & group2 == "t5to10") %>%   pull(statistic)`, \emph{p} = `r t_test_eEPSCs %>%   filter(Treatment == "HNMPA" & group2 == "t5to10") %>%   pull(p_string)`; Figure \ref{summary-plot-HNMPA}) even with HNMPA present. Since DMH neurons show the same response to insulin with HNMPA relative to control conditions, this suggested that insulin acts on DMH neurons through a different receptor.

```{r plot-HNMPA}
#| fig.dim = c(8,5),
#| fig.cap = "Insulin significantly decreases current amplitude over time in both sexes, even when the insulin receptor blocker HNMPA is applied. This suggests that insulin may not require insulin receptors to bind to DMH neurons and influence excitatory synaptic transmission. Each point represents the mean ± SE eEPSC amplitude and \\emph{n} represents the number of unique cells. The asterisks indicate a statistically significant decrease in current amplitude relative to the baseline (t-test; * = p < 0.05, ** = p < 0.01, *** = p < 0.001). The representative traces consist of eEPSCs from one cell averaged over the baseline period (0 to 5 min) and last interval (20 to 25 min) of the recording, and the scale bar represents 50 pA/20 ms. \\label{summary-plot-HNMPA}",
#| fig.scap = "Insulin signalling at DMH neurons is not dependent on insulin receptors",
#| out.width = "100%"

make_pruned_summary_plots(category = 2,
                          treatment = "HNMPA",
                          data = pruned_df,
                          color_choice = 5,
                          representative_trace = male_HNMPA_trace)
```

Insulin-like growth factor receptors have many structural similarities to insulin receptors and they are also responsive to insulin [@nagaoetal2021]. I performed the evoked currents protocol in the presence of an insulin-like growth factor 1 receptor antagonist (PPP). With PPP present, insulin did not significantly decrease evoked current amplitudes over time, even 20 minutes after adding insulin (\emph{t}(`r t_test_eEPSCs %>%   filter(Treatment == "PPP" & group2 == "t5to10") %>%   pull(df)`) = `r t_test_eEPSCs %>%   filter(Treatment == "PPP" & group2 == "t5to10") %>%   pull(statistic)`, \emph{p} = `r t_test_eEPSCs %>%   filter(Treatment == "PPP" & group2 == "t5to10") %>%   pull(p_string)`). However, since the changes in evoked current amplitudes over time with PPP was still not significantly different than the other treatments (Treatment\*Times, \emph{p} = `r tidy_manovar$'p-value'[6]`; Figure \ref{eEPSC-treatment-comparisons-plot}), this suggests that insulin is not fully dependent on insulin-like growth factor 1 receptors to bind to DMH neurons.

```{r plot-PPP}
#| fig.dim = c(8,5),
#| fig.cap = "Insulin does not significantly decrease current amplitude over time with PPP present. PPP is an insulin-like growth factor 1 receptor antagonist. Since blocking these receptors reduced the changes in excitatory synaptic transmission seen in control recordings, insulin may require insulin-like growth factor 1 receptors to act on DMH neurons. Each point represents the mean ± SE eEPSC amplitude and \\emph{n} represents the number of unique cells. There are no asterisks because the decrease in eEPSC amplitude was not statistically significant relative to the baseline. The representative traces consist of eEPSCs from one cell averaged over the baseline period (0 to 5 min) and last interval (20 to 25 min) of the recording, and the scale bar represents 50 pA/20 ms. \\label{summary-plot-PPP}",
#| fig.scap = "Insulin may bind to insulin-like growth factor 1 receptors to influence excitatory synaptic transmission at DMH neurons",
#| out.width = "100%"
make_pruned_summary_plots(category = 2,
                         treatment = "PPP",
                         data = pruned_df,
                         color_choice = 2,
                         representative_trace = female_PPP_trace)
```

## Feeding state does not affect insulin signalling in DMH neurons

Since insulin and DMH neurons are both involved with appetite and digestion [@demeyts2000; @daltonetal1981], I measured evoked excitatory currents over time in rats who had fasted 24 hours before brain removal. Hunger increases the activity of DMH neurons [@angeles-castellanosetal2004], and it is also associated with lowered insulin levels in the bloodstream, which may alter insulin signalling in DMH neurons. Within the first 5 minutes of insulin exposure, evoked current amplitudes in the DMH neurons of fasted rats decreased significantly (\emph{t}(`r t_test_eEPSCs %>%   filter(Treatment == "Fasting" & group2 == "t5to10") %>%   pull(df)`) = `r t_test_eEPSCs %>%   filter(Treatment == "Fasting" & group2 == "t5to10") %>%   pull(statistic)`, \emph{p} = `r t_test_eEPSCs %>%   filter(Treatment == "Fasting" & group2 == "t5to10") %>%   pull(p_string)`; Figure \ref{summary-plot-fasting}).

```{r plot-Fasting}
#| fig.dim = c(8,5),
#| fig.cap = "Insulin significantly decreases current amplitude over time in both sexes after a 24-hour fasting protocol. This suggests that insulin signalling in the DMH is associated with processes beyond appetite regulation. Each point represents the mean ± SE eEPSC amplitude and \\emph{n} represents the number of unique cells. The asterisks indicate a statistically significant decrease in current amplitude relative to the baseline (t-test; * = p < 0.05, ** = p < 0.01, *** = p < 0.001). The representative traces consist of eEPSCs from one cell averaged over the baseline period (0 to 5 min) and last interval (20 to 25 min) of the recording, and the scale bar represents 50 pA/20 ms. \\label{summary-plot-fasting}",
#| fig.scap = "Insulin signalling at the excitatory synapses of DMH neurons is independent of feeding state",
#| out.width = "100%"
make_pruned_summary_plots(category = 2,
                         treatment = "Fasting",
                         data = pruned_df,
                         color_choice = 3,
                         representative_trace = male_fasting_trace)
```

## Synaptic transmission is likely affected by a postsynaptic mechanism

To further explore the mechanism of insulin signalling at the DMH, I used a paired t-test to quantify changes in the PPR over time (Figure \ref{ppr-comparisons-plot}). I used PPRs that were averaged per cell over two five-minute intervals: the baseline (0 to 5 minutes) and the last interval of the recording (20 to 25 minutes), which corresponded to 20 minutes of continuous insulin exposure. The PPR did not change significantly relative to baseline values in the control group (\emph{t}(`r t_test_PPR %>%   filter(Treatment == "Control") %>%   pull(df)`) = `r t_test_PPR %>%   filter(Treatment == "Control") %>%   pull(statistic)`, \emph{p} = `r t_test_PPR %>%   filter(Treatment == "Control") %>%   pull(p_string)`), fasted group (\emph{t}(`r t_test_PPR %>%   filter(Treatment == "Fasting") %>%   pull(df)`) = `r t_test_PPR %>%   filter(Treatment == "Fasting") %>%   pull(statistic)`, \emph{p} = `r t_test_PPR %>%   filter(Treatment == "Fasting") %>%   pull(p_string)`), or PPP-treated cells (\emph{t}(`r t_test_PPR %>%   filter(Treatment == "PPP") %>%   pull(df)`) = `r t_test_PPR %>%   filter(Treatment == "PPP") %>%   pull(statistic)`, \emph{p} = `r t_test_PPR %>%   filter(Treatment == "PPP") %>%   pull(p_string)`). Since the PPR indicates the probability of neurotransmitter release from presynaptic cells, and there was no significant change in PPR with insulin, insulin binding to DMH neurons likely activates postsynaptic signalling pathways.

In the HNMPA treatment group, the paired pulse ratio increased significantly relative to the baseline (\emph{t}(`r t_test_PPR %>%   filter(Treatment == "HNMPA") %>%   pull(df)`) = `r t_test_PPR %>%   filter(Treatment == "HNMPA") %>%   pull(statistic)`, \emph{p} = `r t_test_PPR %>%   filter(Treatment == "HNMPA") %>%   pull(p_string)`), suggesting that preventing insulin from binding to insulin receptors may activate an alternative presynaptic mechanism that ultimately results in the same decrease in excitatory synaptic transmission.

```{r PPR-data-final-summary-plot, warning=F, message=F}
#| fig.cap = "The paired pulse ratio (PPR) did not change significantly after insulin exposure for most of the treatment groups. The change in the PPR relative to the baseline was non-significant for the control, fasting, and PPP treatment groups, but it was statistically significant for the HNMPA-treated cells. Each dot represents the mean PPR of a single cell averaged over the baseline period (0 to 5 minutes) or the last 20 to 25 minutes of the evoked currents protocol. The asterisks indicate a statistically significant difference in the PPR between the two intervals (t-test; * = p < 0.05, ** = p < 0.01, *** = p < 0.001). The solid black circle and triangle overlay represents the mean PPR (± SE) per sex. \\label{ppr-comparisons-plot}",
#| fig.scap = "The paired pulse ratio did not change significantly after insulin exposure for most treatment groups",
#| out.width = "100%"

ppr_summary_plot <- PPR_df %>% 
  mutate(
    Treatment_and_state = paste(Treatment, State, sep = "_"),
    Treatment_and_state = factor(
      Treatment_and_state,
      levels = c(
        "Control_Baseline",
        "Control_Insulin",
        "Fasting_Baseline",
        "Fasting_Insulin",
        "HNMPA_Baseline",
        "HNMPA_Insulin",
        "PPP_Baseline",
        "PPP_Insulin"
      )
    )
  ) %>%
  group_by(Treatment_and_state, Treatment, State, Letter, Sex) %>%
  summarise(mean_PPR_cell = mean(PPR)) %>%
  ungroup() %>%
  ggplot(aes(x = interaction(State,Treatment, sep = "!"), y = mean_PPR_cell, color = Treatment, shape = Sex)) +
  geom_point(size = 2, position = position_jitter(0.04), alpha = 0.9) +
  geom_line(aes(group = Letter), linewidth = connecting_line_width_PPR, alpha = 0.3) +
  stat_summary(
    fun.data = "mean_se",
    geom = "pointrange",
    color = mean_point_colour,
    size = mean_point_size + 0.2,
    alpha = 0.7
  ) +
  scale_color_manual(values = c(my_colours[1],
                                my_colours[3],
                                my_colours[5],
                                my_colours[2])) +
  theme(legend.position = "bottom") +
  scale_x_discrete(guide = guide_axis_nested(delim = "!")) +
  coord_cartesian(ylim = c(0,3)) +
  theme(
    ggh4x.axis.nestline = element_line(color = my_colours_pale[6]),
    axis.text.x = element_text(margin = margin(b = 5, t = 5))
  ) +
  add_bracket(treatment = "Control", y_position = 2.5) +
  add_bracket(treatment = "Fasting", y_position = 2.5) +
  add_bracket(treatment = "HNMPA", y_position = 2.5) +
  add_bracket(treatment = "PPP", y_position = 2.5) +
  labs(
    x = NULL,
    y = "Paired pulse ratio",
    shape = NULL
  ) +
  scale_shape_manual(values = c(17, 16)) +
  guides(color = "none", shape = guide_legend(reverse = T))

if (save_choice == "yes") {
  ggsave(
    plot = ppr_summary_plot,
    filename = "PPR-Summary-Plot-paired.png",
    path = here("Figures/Output-summary-plots"),
    width = 7,
    height = 5,
    units = "in",
    dpi = 300
  )
}

ppr_summary_plot
```

```{r}
kable_t_test_table_PPR
```

```{r treatment-comparisons-plot}
#| fig.cap = "Insulin significantly decreased eEPSC amplitudes in all treatments, with no significant differences between treatments. In this summary figure, each dot represents the mean eEPSC amplitude of a cell relative to the baseline (%) during the last 20-25 minutes of the evoked currents protocol. Black solid shapes indicate the mean ± SE for each sex. \\label{eEPSC-treatment-comparisons-plot}",
#| fig.scap = "Insulin decreased eEPSC amplitudes in all treatments",
#| out.width = "100%"

# ----------------------- Final treatment comparisons summary plot ----------------------------
treatment_comparisons_plot <- cell_coordinates %>%
  group_by(Treatment) %>%
  mutate(
    Treatment = factor(Treatment, levels = list_of_treatments),
    mean_change = mean(percent_change),
    sd_change = sd(percent_change)
  ) %>%
  ggplot(aes(
    x = Treatment,
    y = percent_change,
    color = Treatment,
    shape = Sex
  )) +
  geom_sina(
    bw = 12,
    alpha = 0.8,
    maxwidth = 0.5,
    size = 2
  ) +
  scale_color_manual(values = c(my_colours[1], my_colours[3], my_colours[5], my_colours[2])) +
  scale_shape_manual(values = c(17, 16)) +
  stat_summary(
    fun.data = mean_se,
    geom = "pointrange",
    color = mean_point_colour,
    size = mean_point_size + 0.2,
    alpha = 0.8
  ) +
  geom_hline(yintercept = 100, linetype = "dashed") +
  labs(y = "Change in eEPSC amplitude\n25 min after insulin (%)") +
  theme(legend.position = "right",
        axis.title.y = element_text(angle = 90, hjust = 0.5)) +
  guides(color = "none", shape = guide_legend(reverse = T))

treatment_comparisons_plot

if (save_choice == "yes") {
  ggsave(
    treatment_comparisons_plot,
    path = here("Figures", "Output-summary-plots"),
    file = "Treatment-comparison-plot-percent-change-sex.png",
    width = 7,
    height = 5,
    units = "in",
    dpi = 300
  )
}
```

## Neurons with similar insulin sensitivity are not clustered

Neurons within the DMH are not identical. Depending on their location, they can express receptors for different neurotransmitters and receive projections from different brain regions [@baileyetal2003; @horst&luiten1986; @thompson&swanson1998]. To see if any sub-populations within the DMH were more sensitive to insulin, I analyzed the spatial distribution of the cells in the study. I plotted the cells on a coordinate system and coloured them according to the percent decrease in evoked current amplitude 20 minutes of insulin exposure (Figure \ref{cell-locations}). Neurons with similar responses to insulin did not cluster together, indicating that there are no regions within the DMH that have a greater specificity to insulin signalling compared to the rest of the DMH.

```{r plot-cell-coordinates}
#| fig.cap = "The neurons that I recorded were located in the DMH, and most neurons experienced a decrease in evoked current amplitudes (eEPSC) relative to baseline amplitudes. Each neuron is coloured according to the relative decrease in eEPSC amplitude after 20 minutes of insulin exposure. Neurons with similar responses to insulin did not cluster together, and there were no regions within the DMH that had a greater proportion of highly responsive neurons. In this figure (magnified 5X), the top of the third ventricle is located at (0,0). \\label{cell-locations}",
#| fig.scap = "DMH neurons with similar insulin sensitivity are not clustered together",
#| out.width = "100%"

# ---------------- Plot cell locations ------------------------------------------------------------------------
# Using rasterGrob to insert the DMH slice image as a background picture
# The scale_colour_viridis_c() is only relevant if there are multiple categories
# If you want to show multiple categories, go to geom_point() and remove 'color = my_colours[4]'

cell_coordinates_plot <-
  ggplot(cell_coordinates %>% filter(!is.na(percent_change)), aes(x = X, y = Y, color = percent_change)) +
  annotation_custom(
    rasterGrob(background_slice_image,
      width = unit(1, "npc"),
      height = unit(1, "npc")
    ),
    -Inf, 
    Inf,
    -Inf,
    Inf
  ) +
  labs(y = "y (µM)", x = "x (µM)", color = "% Change in\ncurrent amplitude\nafter 20 min\nof insulin exposure") +
  geom_point(size = 2.7,
             alpha = .99,
             shape = 16) +
  scale_color_viridis_c(
    limits = c(0, max_percent_change),
    begin = 0.27,
    end = 1,
    option = "plasma"
  ) +
  coord_fixed(ratio = 1) +
  xlim(0, 800) +
  scale_y_reverse(limits = c(800, 0)) +
  theme(
    plot.margin = margin(25, 0.5, 0.5, 15),
    axis.title.y = element_text(angle = 90, margin = margin(r = 17)),
    legend.title = element_text(size = 14, margin = margin(b = 15))
  ) +
  guides(color = guide_colorbar(draw.ulim = T, draw.llim = F, theme = theme(
    legend.key.width = unit(0.7, "lines"),
    legend.key.height = unit(10, "lines")
  )))


cell_coordinates_plot

# Save plot
if (save_choice == "yes") {
  ggsave(
    plot = cell_coordinates_plot,
    filename = "Cell_coordinates_plot_add_insulin.png",
    path = here("Figures/Cell_coordinates_plot"),
    width = 14,
    height = 10,
    units = "in",
    dpi = 300,
    scaling = 1.8
  )
}
```

\cleardoublepage

# Discussion

Insulin is a hormone that is closely associated with food intake, and the DMH is a critical brain region for appetite regulation. Although DMH neurons have receptors for insulin, we know little about the effect of insulin on neuronal activity in the DMH. I compared action potential properties before and after adding insulin to see if insulin affects the intrinsic excitability of DMH neurons. I also recorded current amplitudes in DMH neurons both before and after adding insulin to examine the effect of insulin on excitatory synaptic transmission. Overall, insulin significantly decreased both action potential frequency and evoked current amplitude in DMH neurons. These findings support the hypothesis and predictions of the study, with many implications for future studies on insulin signalling within the DMH.

## How does insulin affect neuronal excitability?

After 20 minutes of insulin exposure, DMH neurons fired fewer action potentials, and of the action potentials that did fire, the amplitude was lower. Action potential frequency and amplitude are indicators of intrinsic excitability [@bean2007], so these results would suggest that DMH neurons are less excitable with insulin present. However, the action potential threshold decreased after insulin exposure. Since threshold is the depolarization required to induce an action potential, a lower threshold means that a neuron can more easily become depolarized to the threshold required to fire an action potential, suggesting an increase in intrinsic excitability. These findings are unexpected, as they indicate opposing changes in intrinsic excitability after insulin exposure.

One potential explanation for this effect comes from the significant increase in half-width that occurred after insulin exposure. Half-width is a measure of the repolarization rate [@bean2007], so a slower repolarization rate could indicate changes in action potential kinetics that decrease the neuron's ability to fire action potentials. This may include a decrease in the function of A-type potassium ion channels following an action potential [@kimetal2005]. The K~v~4 potassium channel in particular has been shown to play a critical role in regulating action potential repolarization in hippocampal neurons [@kimetal2005]. However, since the after-polarization amplitude and time of the after-hyperpolarization showed no significant changes, insulin is not likely delaying potassium influx. Further research is required to determine the mechanism leading to decreased AP firing despite a lowered action potential threshold.

## How does insulin affect neuronal communication?

After being exposed to insulin, DMH neurons showed a significant decrease in evoked current amplitudes at glutamate synapses. This finding shows that insulin decreased excitatory synaptic transmission between DMH neurons. A decrease in excitatory synaptic transmission between DMH neurons could affect neuronal pathways associated with processes like energy metabolism. Insulin signalling likely does not affect neuronal pathways involved with appetite regulation, since changes in excitatory synaptic transmission were independent of feeding state.

I found that evoked current amplitudes continued to decrease even in the presence of the receptor blocker HNMPA. Since blocking insulin receptors did not change the decrease in evoked current amplitude seen in the control group, insulin likely does not use insulin receptors to bind to DMH neurons. When PPP (an insulin-like growth factor 1 receptor antagonist) was present, the evoked current amplitudes did not decrease significantly after insulin exposure. Since blocking insulin-like growth factor 1 receptors removed the decrease in excitatory synaptic transmission seen in the control group, this suggests that insulin requires insulin-like growth factor 1 receptors to influence synaptic transmission in the DMH.

It is unclear why insulin would preferentially bind to insulin-like growth factor 1 receptors instead of its own receptor on DMH neurons. And since insulin receptors and insulin-like growth factor 1 receptors share many of the same downstream signalling pathways [@demeyts2000], this does not currently suggest that any specific receptor substrate is favoured over another. The age of the animals in the study could provide a possible explanation for this difference. Insulin-like growth factor 1 signalling has been shown to play a critical role in postnatal brain development (\< 2 months) [@becketal1995], which includes the young rats in this study. Insulin-like growth factor 1 receptors may have exhibited a higher affinity to insulin than insulin receptors because of the importance of insulin-like growth factor 1 receptor activation to brain development.

## What is the mechanism of insulin signalling in the DMH?

The paired pulse ratio was unchanged after insulin exposure for the control, fasting and PPP groups, indicating that insulin likely decreases excitatory synaptic transmission through a post-synaptic mechanism. One potential mechanism is the recruitment of mechanistic target of rapamycin (mTOR) through the phosphatidylinositol 3-kinase (P13K) pathway. mTOR is a serine/threonine protein kinase that has been shown to regulate excitatory synaptic transmission [@mccabeetal2020]. mTOR regulates the number of AMPA receptors expressed on the cell surface [@wangetal2006], which then affects the ability of a cell to respond to glutamate. If mTOR were to reduce the number of AMPA receptors, this would decrease excitatory synaptic transmission since there would be fewer receptors available for glutamate to bind to.

A second post-synaptic mechanism of decreased excitatory synaptic transmission could be changes in the number of active ion channels on cell surfaces [@omalleyetal2003]. In the hippocampus, insulin can inhibit neuronal activity through the activation of ATP-sensitive potassium (K~ATP~) channels [@omalleyetal2003], and there is evidence that there may be a body-weight-dependent mechanism of insulin/K~ATP~ signalling in hypothalamic neurons [@spanswicketal2000]. K~ATP~ channels are activated through the P13K pathway, which can be activated through both insulin receptors and insulin-like growth factor 1 receptors [@demeyts2000].

When HNMPA was present, the paired pulse ratio increased significantly after insulin exposure. This suggests that blocking insulin receptors causes insulin to use a pre-synaptic mechanism to decrease evoked current amplitudes. This may have been through the activation of the mTORC2 complex, which has been shown to decrease neurotransmitter release through pre-synaptic mechanisms [@mccabeetal2020].

<!-- Although mTOR is present in either the mTORC1 or mTORC2 complexes, which each have distinctive roles in altering synaptic transmission [@anglikeretal2015], both forms are required for neuronal function.  -->

In all treatments, DMH neurons responded similarly to insulin in males and females. When observing changes in evoked current amplitudes, I found no significant differences between the sexes over time. These findings suggest that there are no sex-based differences in the intracellular pathways affected by insulin. However, this does not mean that insulin signalling is completely independent of sex. All rats were younger than 44 days old, which is less than the average onset of puberty [@mccutcheon&marinelli2009]. At this age, many sex-based differences are not yet present. Experiments with older animals are required before any conclusive sex-based differences can be determined.

## What are the next steps?

We now know that insulin decreases excitatory current amplitudes in DMH neurons, which provides opportunities for research on insulin signalling in the DMH. The goal of future studies is to learn which specific processes insulin initiates when binding to DMH neurons, and how this affects neuronal communication, and ultimately physiology. As the concentration of insulin used in this study is much higher than hypothalamic insulin levels (approximately 0.16 nM) [@baskinetal1983], future research could examine if using physiologically relevant insulin concentrations results in the same changes in neuronal activity. Future studies could also use a wider range of insulin concentrations to determine the optimal insulin concentrations required to modify synaptic transmission and neuronal excitability.

The insulin receptor blocker experiments suggested that insulin-like growth factor 1 receptors are responsible for insulin binding in the DMH. Recordings conducted in the presence of both receptor antagonists could confirm if insulin binds primarily to insulin-like growth factor 1 receptors, insulin receptors, a mixture of both receptors, or a novel receptor not directly associated with insulin. Since each of these receptors is associated with different pathways within cells [@demeyts2000], identifying the binding mechanism of insulin on DMH neurons could provide insight into the specific processes that insulin uses to influence neuronal communication. As mTOR is an important potential source of insulin-signalling in the DMH, applying antagonists of the P13K pathway, mTOR and its downstream effectors could provide information on the specific intracellular processes that influence neuronal activity and synaptic transmission.

There are also opportunities for whole-animal studies, such as injecting insulin into the DMH and monitoring changes in body weight and energy metabolism. This study raises new questions about the effects of health on insulin signalling in the DMH. The rats used in this study were young and healthy. Could conditions like diabetes, which affects insulin levels in the body, affect DMH neurons? Would DMH neurons respond differently in animals with health conditions like obesity? How would insulin signalling in the DMH change in older animals?

## Conclusions

Insulin acts in the DMH to decrease the amount of neuronal communication between neurons and alter intrinsic excitability. It achieves this response by binding to a receptor (likely an insulin-like growth factor 1 receptor), to activate an intracellular pathway that decreases excitatory synaptic transmission and action potential frequency. These findings contribute to research on the neurophysiological mechanisms of energy metabolism, with many implications for health research.

Exploring insulin signalling within the DMH also contributes to our understanding of the actions of insulin in the brain, and its involvement with energy metabolism. These findings have direct applications to understanding the effects of health conditions like diabetes, where the brain experiences reduced insulin levels and/or impaired insulin signalling [@blazquezetal2014]. Learning more about the interactions between insulin and DMH neurons could also provide insight into long-term changes in appetite and energy metabolism associated with obesity, with the possibility of contributing to the development of new obesity therapies.

\cleardoublepage

# Acknowledgements

I would like to thank my honours supervisor, Dr. Karen Crosby, who guided me throughout every step of this research project, listened carefully to my ideas, and taught me how to become an independent researcher.

I would also like to thank my committee member, Dr. Emily Austen, for her thoughtful feedback on this project and for giving me new ideas on statistical modelling.

Thank you to Jackie Jacob-Vogels for her close attention to detail while looking after the rats.

Kate Weldon and Selena Winchester were wonderful lab volunteers who made it possible to complete experiments during the semester.

Jasmine Pannozzo-Oddy, a 2024 Independent Studies student, has had a significant impact on what I learned during this project through her work in exploring the effect of insulin on *inhibitory* synaptic transmission in the DMH.

Thank you especially to my family, who listened (very, very) patiently to many variations of my presentations and remained enthusiastic at all stages of the research project. I'd like to thank my Aunt, Linda Fraser, in particular, who waited for me after long hours in the lab and frequently sat down with me to ask me about my project. And of course, my Mom, for everything she has done.

Thank you to Ruby, and all the rats.

\vspace*{\fill}

```{=tex}
\begin{centering}

\adforn{17}

\end{centering}
```
*This research was partly funded by a grant from the Natural Sciences and Engineering Research Council of Canada (NSERC) held by my supervisor. I also received funding through the Research New Brunswick Summer Studentship (2023).*

\cleardoublepage

# References

```{r exit-knitting}
knitr::knit_exit()
```

# ----------- Exit Knitting ------------------------

```{r extract-r-code, include=FALSE}
# ---------------- Write R chunks to a script file ---------------------------------------------
# This code extracts code from Thesis.Rmd and saves it to Scripts/Scripts-from-Thesis-Rmd.R
# Useful if you want an independent script file for debugging

# I set documentation = 0 to extract R code within chunks only
# This code is commented out because it causes issues with knitting
# R thought that there were duplicate chunks

# knitr::purl(input = here::here("Thesis/Thesis.Rmd"), output = here::here("Scripts/Scripts-from-Thesis-Rmd.R"), documentation = 0)
```

# Get Presentation Plots

```{r get-presentation-plot-no-treatment, eval=F, purl=F, fig.width = 8}
# -------------------------- Get Presentation Plots

# Title is more readable, and there are significance asterisks
# The significance asterisks come from the t_test_eEPSCs table generated
# from the statistical models earlier in this script


# Presentation plot for no treatment condition
presentation_plot_no_treatment <-
  make_presentation_plots(category = 2,
                         treatment = "Control",
                         data = pruned_df,
                         color_choice = 1) +
  labs(title = "Insulin decreases eEPSC amplitudes in <b style='color:#4d0299'>males</b> & <b style='color:#b071f0;'>females</b> by 50%")

presentation_plot_no_treatment

if (save_choice == "yes") {
  ggsave(
    presentation_plot_no_treatment,
    path = here("Figures/Output-summary-plots"),
    file = "Presentation-Plot-No-Treatment.png",
    width = 12,
    height = 7,
    units = "in",
    dpi = 300,
    scaling = 1.5
  )
}
```

# No Treatment, Males

```{r one-sex-only-plots}
male_point_shape <- 16
female_point_shape <- 15
male_point_color <- my_colours_very_dark
female_point_color <- my_colours_very_pale


# Presentation plot for no treatment condition
presentation_plot_no_treatment_FEMALES <-
  make_presentation_plots_one_sex(
    sex = "Female",
    category = 2,
    point_shape = female_point_shape,
    treatment = "Control",
    data = pruned_df,
    point_color = female_point_color[1]
  ) +
  # For Dr. Crosby's Presentation
  theme(axis.title = element_text(face = "plain"),
        axis.title.y = element_text(angle = 90))

presentation_plot_no_treatment_FEMALES

presentation_plot_no_treatment_MALES <-
  make_presentation_plots_one_sex(
    sex = "Male",
    category = 2,
    point_shape = male_point_shape,
    treatment = "Control",
    data = pruned_df,
    point_color = male_point_color[1]
  ) +
  theme(axis.title = element_text(face = "plain"),
        axis.title.y = element_text(angle = 90))
  
presentation_plot_no_treatment_MALES

save_choice <- "no"
if (save_choice == "yes") {
  ggsave(
    presentation_plot_no_treatment_FEMALES,
    path = here("Figures/Output-summary-plots"),
    file = "Presentation-Plot-No-Treatment-FEMALES.png",
    width = 12,
    height = 7,
    units = "in",
    dpi = 300,
    scaling = 1.5
  )
  
  ggsave(
    presentation_plot_no_treatment_MALES,
    path = here("Figures/Output-summary-plots"),
    file = "Presentation-Plot-No-Treatment-MALES.png",
    width = 12,
    height = 7,
    units = "in",
    dpi = 300,
    scaling = 1.5
  )
}

save_choice <- "no"
```

```{r get-presentation-plot-HNMPA, eval=F, purl=F}
# Presentation plot for HNMPA
# I used orange colours to match with the presentation illustrations

presentation_plot_HNMPA <- make_presentation_plots(category = 2,
                                                  treatment = "HNMPA",
                                                  data = pruned_df,
                                                  color_choice = 5) +
  labs(title = "Insulin decreases eEPSC amplitudes even with HNMPA")

presentation_plot_HNMPA

if (save_choice == "yes") {
  ggsave(
    presentation_plot_HNMPA,
    path = here("Figures/Output-summary-plots"),
    file = "Presentation-Plot-HNMPA.png",
    width = 12,
    height = 7,
    units = "in",
    dpi = 300,
    scaling = 1.5
  )
}
```

```{r get-presentation-plot-fasting, eval=F, purl=F}
# Presentation plot for fasting

presentation_plot_fasting <- make_presentation_plots(category = 2,
                                                    treatment = "Fasting",
                                                    data = pruned_df,
                                                    color_choice = 3) +
  labs(title = "Insulin decreases eEPSC amplitudes in fasted <b style='color:#398511'>males</b> & <b style='color:#92d46e'>females</b>")


# +
# theme(
#   axis.title = element_text(face = "plain"),
#   axis.title.y = element_text(angle = 90)
# )

presentation_plot_fasting

if (save_choice == "yes") {
  ggsave(
    presentation_plot_fasting,
    path = here("Figures/Output-summary-plots"),
    file = "Presentation-Plot-Fasting.png",
    width = 12,
    height = 7,
    units = "in",
    dpi = 300,
    scaling = 1.5
  )
}
```

```{r get-presentation-plot-PPP, eval=F, purl=F}
# Presentation plot for PPP

presentation_plot_PPP <- make_presentation_plots(category = 2,
                                                treatment = "PPP",
                                                data = pruned_df,
                                                color_choice = 2) +
  labs(title = "eEPSC amplitudes do not significantly decrease with PPP")

# +
# theme(
#   axis.title = element_text(face = "plain"),
#   axis.title.y = element_text(angle = 90)
# )

presentation_plot_PPP

if (save_choice == "yes") {
  ggsave(
    presentation_plot_PPP,
    path = here("Figures/Output-summary-plots"),
    file = "Presentation-Plot-PPP.png",
    width = 12,
    height = 7,
    units = "in",
    dpi = 300,
    scaling = 1.5
  )
}
```

```{r cell-coordinates-no-axes, eval=F, purl = F}
cell_coordinates_no_axes_2 <-
  ggplot(cell_coordinates %>% filter(!is.na(percent_change)), aes(x = X, y = Y, fill = percent_change)) +
  annotation_custom(
    rasterGrob(background_slice_image,
      width = unit(1, "npc"),
      height = unit(1, "npc")
    ),
    -Inf, 
    Inf,
    -Inf,
    Inf
  ) +
  geom_point(size = 5,
             alpha = 0.9,
             shape = 21,
             color = "white",
             stroke = 1.3) +
  scale_fill_viridis_c(
    limits = c(0, max_percent_change),
    begin = 0.27,
    end = 1,
    option = "plasma"
  ) +
  coord_fixed(ratio = 1) +
  xlim(0, 800) +
  scale_y_reverse(limits = c(800, 0)) +
  labs(
    y = NULL,
    x = NULL,
    fill = "% Change in\ncurrent amplitude\nafter 20 min of insulin exposure"
  ) +
  theme(
    plot.margin = margin(0, 0, 0, 0),
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.title = element_text(size = 14, face = "bold", margin = margin(b = 15))) +
  guides(fill = guide_colorbar(draw.ulim = T, draw.llim = F, theme = theme(
    legend.key.width = unit(1, "lines"),
    legend.key.height = unit(15, "lines"),
    legend.title = element_text(size = 14, face = "bold", margin = margin(b = 15))
  )))


cell_coordinates_no_axes_2


ggsave(
    plot = cell_coordinates_no_axes_2,
    filename = "Cell_coordinates_no_axes.png",
    path = here("Figures/Cell_coordinates_plot"),
    width = 14,
    height = 10,
    units = "in",
    dpi = 300,
    scaling = 1.8
  )
```

```{r cell-coordinates-no-axes-by-treatment, eval = F, purl = F}
cell_coordinates_no_axes_by_treatment <-
  ggplot(cell_coordinates %>% filter(!is.na(percent_change)), aes(x = X, y = Y, fill = Treatment)) +
  annotation_custom(
    rasterGrob(background_slice_image,
      width = unit(1, "npc"),
      height = unit(1, "npc")
    ),
    -Inf, 
    Inf,
    -Inf,
    Inf
  ) +
  geom_point(size = 5,
             alpha = 0.9,
             shape = 21,
             color = "white",
             stroke = 1.3) +
  scale_fill_viridis_d(
    begin = 0.27,
    end = 1,
    option = "plasma"
  ) +
  coord_fixed(ratio = 1) +
  xlim(0, 800) +
  scale_y_reverse(limits = c(800, 0)) +
  labs(
    y = NULL,
    x = NULL,
    fill = "% Change in\ncurrent amplitude\nafter 20 min"
  ) +
  theme(
    plot.margin = margin(0, 0, 0, 0),
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.title = element_text(size = 14, face = "bold", margin = margin(b = 15)),
    legend.key.size = unit(0.5, 'in'),
    legend.text = element_text(size = 12)
  )
cell_coordinates_no_axes_by_treatment

if (save_choice == "yes") {
  ggsave(
    plot = cell_coordinates_no_axes_by_treatment,
    filename = "Cell_coordinates_no_axes_by_treatment.png",
    path = here("Figures/Cell_coordinates_plot"),
    width = 14,
    height = 10,
    units = "in",
    dpi = 300,
    scaling = 1.8
  )
}
```

# Get AP Trace for schematic

```{r get-representative-current-clamp-steps-image}
all_AP_steps_plot <- abf2_load("../Data/ABF-Files/23d15001.abf") %>%
  MeltAbf() %>%
  rename("Voltage" = chan1,
         "Current" = chan2) %>%
  mutate(
    Sweep = str_sub(Episode, 4),
    Sweep = factor(Sweep, levels = c(
      "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"
    )),
    Current_injection = case_when(
      Sweep == 1 ~ -50,
      Sweep == 2 ~ -40,
      Sweep == 3 ~ -30,
      Sweep == 4 ~ -20,
      Sweep == 5 ~ -10,
      Sweep == 6 ~ 0,
      Sweep == 7 ~ 10,
      Sweep == 8 ~ 20,
      Sweep == 9 ~ 30,
      Sweep == 10 ~ 40,
      T ~ 50
    ),
    Current_injection = factor(
      Current_injection,
      levels = c(-50,-40,-30 , -20,-10, 0, 10, 20, 30, 40)
    )
  ) %>%
  ggplot(aes(
    x = Time,
    y = Voltage,
    group = Current_injection,
    color = Current_injection
  )) +
  geom_line(linewidth = 0.5) +
  labs(x = NULL,
       y = NULL) +
  coord_cartesian(
    xlim  = c(ap_traces_x_min, ap_traces_x_max+500),
    ylim = c(ap_traces_y_min, ap_traces_y_max)
  ) +
  theme_void() +
  scale_color_viridis_d(begin = 0,
                        end = 0.8,
                        option = "plasma",
                        direction = -1) +
  guides(color = guide_legend(override.aes = list(linewidth = 1.2), reverse = T)) +
  theme(
    text = element_text(family = plot_font_family),
    legend.key.width = unit(0.8, "line"),
    plot.margin = margin(20, 20, 20, 20)
  ) +
  labs(
    color = "Current Injection (pA)"
  ) +
  theme(
    legend.position = "none"
  )

all_AP_steps_plot

ggsave(
    plot = all_AP_steps_plot,
    path = here("Figures", "Representative-Traces"),
    file = "All-AP-steps-plot-23d15001.svg",
    width = 7,
    height = 5,
    units = "in",
    dpi = 300
  )
```

<!-- **Hormone**: A molecule that travels through the bloodstream, or within organs, to change the activity of other cells. Insulin is a hormone that is released by the pancreas to control blood sugar. -->

<!-- **Neurotransmitter**: A small molecule that interacts with a neuron and changes its activity. Neurons communicate by releasing neurotransmitters onto each other. -->

<!-- **Inhibition**: A change within a neuron that decreases its activity and interactions with other neurons. -->

<!-- **GABA**: Short for gamma-aminobutyric acid, this is an inhibitory neurotransmitter. When GABA binds to a neuron, it decreases the likelihood of the neuron communicating with other neurons. -->

<!-- **Excitation**: The opposite of inhibition. When a neuron is excited (for example, by an excitatory neurotransmitter), its activity increases and it interacts with other neurons. -->

<!-- **Glutamate**: An excitatory neurotransmitter that binds to a neuron and increases its activity. -->

<!-- **Ion channel**: A molecule located within a cell membrane that allows specific ions to enter and/or exit the cell. -->

<!-- **Neuronal communication**: The amount of interaction between two or more neurons. -->

<!-- **Artificial cerebrospinal fluid**: Artificial cerebrospinal fluid is a solution containing ions, sugars, and other molecules to mimic the cerebrospinal fluid that surrounds the brain and spinal cord. I used fluid that was continuously bubbled with oxygen to maintain healthy, living brain slices for the recordings. -->

<!-- **ANOVA**: An abbreviation for analysis of variance. This is a statistical test that allows us to quantify the differences between multiple groups. I used a multivariate repeated measures ANOVA to explore changes in current amplitude change over time. Since the variables (in this case, current) are measured repeatedly from the same subject, it is called repeated measures. -->
